<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>13.2 ArchR and Custom Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="13.2 ArchR and Custom Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="13.2 ArchR and Custom Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-05-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="motif-deviations.html"/>
<link rel="next" href="footprinting-with-archr.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://www.archrproject.com/index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li><a href="index.html#section"><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>1.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="1.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>1.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="1.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>1.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>1.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="1.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="1.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>1.7</b> Per-cell Quality Control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>2.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="2.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="3.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>4.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="4.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>4.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="4.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>5.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>6.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="6.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>6.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="6.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>6.3</b> Dimensionality Reduction After Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>7</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>7.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="7.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>7.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="7.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>7.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="7.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>7.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="7.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="7.6" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.6</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.7" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.7</b> Launching the ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>8</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>8.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>9</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="9.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>10.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="10.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>10.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="10.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>10.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="10.4" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>10.4</b> Add Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>11</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>11.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="11.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>11.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="11.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>11.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-enrichment-in-differential-peaks.html"><a href="motif-enrichment-in-differential-peaks.html"><i class="fa fa-check"></i><b>12.1</b> Motif Enrichment in Differential Peaks</a></li>
<li class="chapter" data-level="12.2" data-path="motif-enrichment-in-marker-peaks.html"><a href="motif-enrichment-in-marker-peaks.html"><i class="fa fa-check"></i><b>12.2</b> Motif Enrichment in Marker Peaks</a></li>
<li class="chapter" data-level="12.3" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>12.3</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="12.4" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>12.4</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>13</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>13.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="13.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>13.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>14</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>14.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="14.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>14.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="14.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>14.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>15</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>15.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="15.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>15.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="15.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>15.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="15.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>15.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>16</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="myeloid-trajectory-monocyte-differentiation.html"><a href="myeloid-trajectory-monocyte-differentiation.html"><i class="fa fa-check"></i><b>16.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="16.2" data-path="lymphoid-trajectory-b-cell-cifferentiation.html"><a href="lymphoid-trajectory-b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>16.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>17</b> Plot aesthetics in ArchR</a></li>
<li class="chapter" data-level="18" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>18</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="index.html">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="archr-and-custom-deviations" class="section level2" number="13.2">
<h2><span class="header-section-number">13.2</span> ArchR and Custom Deviations</h2>
<p>In the chapter on Peak Annotation Enrichment, we introduced how to create peak annotations for any set of genomic regions. This encluded (i) ArchR-supported region sets such as curated TF binding sites from ENCODE and peak sets from bulk ATAC-seq and (ii) custom user-supplied region sets. If you have not read this section, we recommend doing so to better understand how peak annotations work.</p>
<p>These peak annotations can be used in deviation calculations in the same way as motifs. Here we provide examples of how to run these analyses but note that the downstream analyses are identical to what was shown in the previous section for motifs and thus we do not provide extensive details on each step of the code. Once you create a deviations matrix in your Arrow files, the rest is the same.</p>
<div id="encode-tfbs" class="section level3" number="13.2.1">
<h3><span class="header-section-number">13.2.1</span> Encode TFBS</h3>
<p>In case you have not added an annotations matrix for the “EncodeTFBS” region set, lets do that now.</p>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="archr-and-custom-deviations.html#cb251-1"></a><span class="cf">if</span>(<span class="st">&quot;EncodeTFBS&quot;</span> <span class="op">%ni%</span><span class="st"> </span><span class="kw">names</span>(projHeme5<span class="op">@</span>peakAnnotation)){</span>
<span id="cb251-2"><a href="archr-and-custom-deviations.html#cb251-2"></a>    projHeme5 &lt;-<span class="st"> </span><span class="kw">addArchRAnnotations</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">collection =</span> <span class="st">&quot;EncodeTFBS&quot;</span>)</span>
<span id="cb251-3"><a href="archr-and-custom-deviations.html#cb251-3"></a>}</span></code></pre></div>
<p>Then, we create a deviations matrix, providing this peak annotation to the <code>peakAnnotation</code> parameter.</p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="archr-and-custom-deviations.html#cb252-1"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addDeviationsMatrix</span>(</span>
<span id="cb252-2"><a href="archr-and-custom-deviations.html#cb252-2"></a>  <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb252-3"><a href="archr-and-custom-deviations.html#cb252-3"></a>  <span class="dt">peakAnnotation =</span> <span class="st">&quot;EncodeTFBS&quot;</span>,</span>
<span id="cb252-4"><a href="archr-and-custom-deviations.html#cb252-4"></a>  <span class="dt">force =</span> <span class="ot">TRUE</span></span>
<span id="cb252-5"><a href="archr-and-custom-deviations.html#cb252-5"></a>)</span></code></pre></div>
<blockquote>
<p>## Using Previous Background Peaks!<br />
## ArchR logging to : ArchRLogs/ArchR-addDeviationsMatrix-10374737762a7-Date-2020-04-15_Time-10-59-20.log<br />
## If there is an issue, please report to github with logFile!<br />
## NULL<br />
## 2020-04-15 10:59:26 : Batch Execution w/ safelapply!, 0 mins elapsed.<br />
## ###########<br />
## 2020-04-15 11:08:11 : Completed Computing Deviations!, 8.864 mins elapsed.<br />
## ###########<br />
## ArchR logging successful to : ArchRLogs/ArchR-addDeviationsMatrix-10374737762a7-Date-2020-04-15_Time-10-59-20.log</p>
</blockquote>
<p>We can create a dot plot of the ranked deviations.</p>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="archr-and-custom-deviations.html#cb253-1"></a>plotVarDev &lt;-<span class="st"> </span><span class="kw">getVarDeviations</span>(projHeme5, <span class="dt">plot =</span> <span class="ot">TRUE</span>, <span class="dt">name =</span> <span class="st">&quot;EncodeTFBSMatrix&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## DataFrame with 6 rows and 6 columns<br />
## seqnames idx name combinedVars<br />
## <Rle> <array> <array> <numeric><br />
## f222 z 222 222.GATA2_S-K562… 13.6382687713858<br />
## f41 z 41 41.EZH2_39-NHEK… 12.9150554355261<br />
## f542 z 542 542.TAL1_SC-K562… 11.5490744327685<br />
## f584 z 584 584.GATA_1-PBDEFetal… 10.7653053700024<br />
## f498 z 498 498.GATA_2-K562… 10.2384173319845<br />
## f44 z 44 44.EZH2_39-NHLF… 9.92292044310219<br />
## combinedMeans rank<br />
## <numeric> <integer><br />
## f222 -0.0312421757573656 1<br />
## f41 0.113221166775704 2<br />
## f542 -0.0155915927183644 3<br />
## f584 -0.00997025022971285 4<br />
## f498 -0.0317577572129743 5<br />
## f44 0.0707460230932926 6</p>
</blockquote>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="archr-and-custom-deviations.html#cb254-1"></a>plotVarDev</span></code></pre></div>
<blockquote>
<p>## Warning: Removed 7 rows containing missing values (geom_point).</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Variable-EncodeTFBS-Deviation-Scores_1.png" width="500" height="500" /></p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="archr-and-custom-deviations.html#cb255-1"></a><span class="kw">plotPDF</span>(plotVarDev, <span class="dt">name =</span> <span class="st">&quot;Variable-EncodeTFBS-Deviation-Scores&quot;</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## Warning: Removed 7 rows containing missing values (geom_point).<br />
## Warning: Removed 7 rows containing missing values (geom_point).<br />
## [1] 0</p>
</blockquote>
<p>Or we can subset these TF binding sites to particular motifs we are interested in and then plot their deviation z-scores per-cell on our UMAP embedding.</p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="archr-and-custom-deviations.html#cb256-1"></a>tfs &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;GATA_1&quot;</span>, <span class="st">&quot;CEBPB&quot;</span>, <span class="st">&quot;EBF1&quot;</span>, <span class="st">&quot;IRF4&quot;</span>, <span class="st">&quot;TBX21&quot;</span>, <span class="st">&quot;PAX5&quot;</span>)</span>
<span id="cb256-2"><a href="archr-and-custom-deviations.html#cb256-2"></a><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(tfs, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;EncodeTFBSMatrix&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “z:584.GATA_1-PBDEFetal…” “z:582.GATA_1-PBDE…”<br />
## [3] “z:497.GATA_1-K562…” “z:477.CEBPB-K562…”<br />
## [5] “z:462.CEBPB-IMR90…” “z:427.CEBPB-HepG2…”<br />
## [7] “z:426.CEBPB-HepG2…” “z:379.CEBPB-HeLa_S3…”<br />
## [9] “z:344.CEBPB-H1_hESC…” “z:293.EBF1_SC-GM12878…”<br />
## [11] “z:278.CEBPB-A549…” “z:213.CEBPB_S-K562…”<br />
## [13] “z:173.CEBPB_S-HepG2…” “z:130.PAX5_C2-GM12892…”<br />
## [15] “z:123.PAX5_C2-GM12891…” “z:102.PAX5_N1-GM12878…”<br />
## [17] “z:101.PAX5_C2-GM12878…” “z:93.IRF4_SC-GM12878…”<br />
## [19] “z:87.EBF1_SC-GM12878…” “z:86.CEBPB_S-GM12878…”<br />
## [21] “deviations:584.GATA_1-PBDEFetal…” “deviations:582.GATA_1-PBDE…”<br />
## [23] “deviations:497.GATA_1-K562…” “deviations:477.CEBPB-K562…”<br />
## [25] “deviations:462.CEBPB-IMR90…” “deviations:427.CEBPB-HepG2…”<br />
## [27] “deviations:426.CEBPB-HepG2…” “deviations:379.CEBPB-HeLa_S3…”<br />
## [29] “deviations:344.CEBPB-H1_hESC…” “deviations:293.EBF1_SC-GM12878…”<br />
## [31] “deviations:278.CEBPB-A549…” “deviations:213.CEBPB_S-K562…”<br />
## [33] “deviations:173.CEBPB_S-HepG2…” “deviations:130.PAX5_C2-GM12892…”<br />
## [35] “deviations:123.PAX5_C2-GM12891…” “deviations:102.PAX5_N1-GM12878…”<br />
## [37] “deviations:101.PAX5_C2-GM12878…” “deviations:93.IRF4_SC-GM12878…”<br />
## [39] “deviations:87.EBF1_SC-GM12878…” “deviations:86.CEBPB_S-GM12878…”</p>
</blockquote>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="archr-and-custom-deviations.html#cb257-1"></a>markerTFs &lt;-<span class="st"> </span><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(tfs, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;EncodeTFBSMatrix&quot;</span>)</span>
<span id="cb257-2"><a href="archr-and-custom-deviations.html#cb257-2"></a>markerTFs &lt;-<span class="st"> </span><span class="kw">sort</span>(<span class="kw">grep</span>(<span class="st">&quot;z:&quot;</span>, markerTFs, <span class="dt">value =</span> <span class="ot">TRUE</span>))</span>
<span id="cb257-3"><a href="archr-and-custom-deviations.html#cb257-3"></a>TFnames &lt;-<span class="st"> </span>stringr<span class="op">::</span><span class="kw">str_split</span>(stringr<span class="op">::</span><span class="kw">str_split</span>(markerTFs, <span class="dt">pattern =</span> <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="dt">simplify=</span><span class="ot">TRUE</span>)[,<span class="dv">2</span>], <span class="dt">pattern =</span> <span class="st">&quot;-&quot;</span>, <span class="dt">simplify =</span> <span class="ot">TRUE</span>)[,<span class="dv">1</span>]</span>
<span id="cb257-4"><a href="archr-and-custom-deviations.html#cb257-4"></a>markerTFs &lt;-<span class="st"> </span>markerTFs[<span class="op">!</span><span class="kw">duplicated</span>(TFnames)]</span>
<span id="cb257-5"><a href="archr-and-custom-deviations.html#cb257-5"></a>markerTFs</span></code></pre></div>
<blockquote>
<p>## [1] “z:101.PAX5_C2-GM12878…” “z:102.PAX5_N1-GM12878…”<br />
## [3] “z:173.CEBPB_S-HepG2…” “z:278.CEBPB-A549…”<br />
## [5] “z:293.EBF1_SC-GM12878…” “z:497.GATA_1-K562…”<br />
## [7] “z:93.IRF4_SC-GM12878…”</p>
</blockquote>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="archr-and-custom-deviations.html#cb258-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb258-2"><a href="archr-and-custom-deviations.html#cb258-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb258-3"><a href="archr-and-custom-deviations.html#cb258-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;EncodeTFBSMatrix&quot;</span>, </span>
<span id="cb258-4"><a href="archr-and-custom-deviations.html#cb258-4"></a>    <span class="dt">name =</span> markerTFs, </span>
<span id="cb258-5"><a href="archr-and-custom-deviations.html#cb258-5"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb258-6"><a href="archr-and-custom-deviations.html#cb258-6"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme5)</span>
<span id="cb258-7"><a href="archr-and-custom-deviations.html#cb258-7"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-103747b681a1c-Date-2020-04-15_Time-11-08-26.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = EncodeTFBSMatrix<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting Embedding<br />
## 1 2 3 4 5 6 7<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-103747b681a1c-Date-2020-04-15_Time-11-08-26.log</p>
</blockquote>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="archr-and-custom-deviations.html#cb259-1"></a>p2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb259-2"><a href="archr-and-custom-deviations.html#cb259-2"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb259-3"><a href="archr-and-custom-deviations.html#cb259-3"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb259-4"><a href="archr-and-custom-deviations.html#cb259-4"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb259-5"><a href="archr-and-custom-deviations.html#cb259-5"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb259-6"><a href="archr-and-custom-deviations.html#cb259-6"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb259-7"><a href="archr-and-custom-deviations.html#cb259-7"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb259-8"><a href="archr-and-custom-deviations.html#cb259-8"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb259-9"><a href="archr-and-custom-deviations.html#cb259-9"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb259-10"><a href="archr-and-custom-deviations.html#cb259-10"></a>    )</span>
<span id="cb259-11"><a href="archr-and-custom-deviations.html#cb259-11"></a>})</span>
<span id="cb259-12"><a href="archr-and-custom-deviations.html#cb259-12"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-EncodeTFBS-W-Imputation_1.png" width="800" height="800" /></p>
</div>
<div id="bulk-atac-seq-1" class="section level3" number="13.2.2">
<h3><span class="header-section-number">13.2.2</span> Bulk ATAC-seq</h3>
<p>Similarly, we can use ArchR-curated bulk ATAC-seq peak sets for our motif deviation calculations.If you have not added motif annotations</p>
<p>In case you have not added an annotations matrix for the “EncodeTFBS” region set, lets do that now.</p>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="archr-and-custom-deviations.html#cb260-1"></a><span class="cf">if</span>(<span class="st">&quot;ATAC&quot;</span> <span class="op">%ni%</span><span class="st"> </span><span class="kw">names</span>(projHeme5<span class="op">@</span>peakAnnotation)){</span>
<span id="cb260-2"><a href="archr-and-custom-deviations.html#cb260-2"></a>    projHeme5 &lt;-<span class="st"> </span><span class="kw">addArchRAnnotations</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">collection =</span> <span class="st">&quot;ATAC&quot;</span>)</span>
<span id="cb260-3"><a href="archr-and-custom-deviations.html#cb260-3"></a>}</span></code></pre></div>
<p>Then, we create a deviations matrix, providing this peak annotation to the <code>peakAnnotation</code> parameter.</p>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="archr-and-custom-deviations.html#cb261-1"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addDeviationsMatrix</span>(</span>
<span id="cb261-2"><a href="archr-and-custom-deviations.html#cb261-2"></a>  <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb261-3"><a href="archr-and-custom-deviations.html#cb261-3"></a>  <span class="dt">peakAnnotation =</span> <span class="st">&quot;ATAC&quot;</span>,</span>
<span id="cb261-4"><a href="archr-and-custom-deviations.html#cb261-4"></a>  <span class="dt">force =</span> <span class="ot">TRUE</span></span>
<span id="cb261-5"><a href="archr-and-custom-deviations.html#cb261-5"></a>)</span></code></pre></div>
<blockquote>
<p>## Using Previous Background Peaks!<br />
## ArchR logging to : ArchRLogs/ArchR-addDeviationsMatrix-103743481be2a-Date-2020-04-15_Time-11-09-46.log<br />
## If there is an issue, please report to github with logFile!<br />
## NULL<br />
## 2020-04-15 11:09:52 : Batch Execution w/ safelapply!, 0 mins elapsed.<br />
## ###########<br />
## 2020-04-15 11:13:09 : Completed Computing Deviations!, 3.376 mins elapsed.<br />
## ###########<br />
## ArchR logging successful to : ArchRLogs/ArchR-addDeviationsMatrix-103743481be2a-Date-2020-04-15_Time-11-09-46.log</p>
</blockquote>
<p>We can create a dot plot of the ranked deviations.</p>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="archr-and-custom-deviations.html#cb262-1"></a>plotVarDev &lt;-<span class="st"> </span><span class="kw">getVarDeviations</span>(projHeme5, <span class="dt">plot =</span> <span class="ot">TRUE</span>, <span class="dt">name =</span> <span class="st">&quot;ATACMatrix&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## DataFrame with 6 rows and 6 columns<br />
## seqnames idx name combinedVars combinedMeans<br />
## <Rle> <array> <array> <numeric> <numeric><br />
## f22 z 22 IAtlas_T_CD8posCenMem 12.8413703976471 -0.0972946574781075<br />
## f86 z 86 Heme_CD8 12.5070480860108 -0.079462324469343<br />
## f85 z 85 Heme_CD4 12.2753260246254 -0.0561621531675016<br />
## f23 z 23 IAtlas_T_CD8posEffMem 12.2250185804317 -0.104389732164243<br />
## f21 z 21 IAtlas_T_CD8pos 12.1998071900673 -0.09178088897301<br />
## f33 z 33 IAtlas_T_Th1Precursor 12.1352640996498 -0.0819315630754632<br />
## rank<br />
## <integer><br />
## f22 1<br />
## f86 2<br />
## f85 3<br />
## f23 4<br />
## f21 5<br />
## f33 6</p>
</blockquote>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="archr-and-custom-deviations.html#cb263-1"></a>plotVarDev</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Variable-ATAC-Deviation-Scores_1.png" width="500" height="500" /></p>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="archr-and-custom-deviations.html#cb264-1"></a><span class="kw">plotPDF</span>(plotVarDev, <span class="dt">name =</span> <span class="st">&quot;Variable-ATAC-Deviation-Scores&quot;</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## [1] 0</p>
</blockquote>
<p>Or we can plot the deviation z-scores for each of these peak sets per-cell on our UMAP embedding.</p>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="archr-and-custom-deviations.html#cb265-1"></a>ATACPeaks &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Heme_HSC&quot;</span>, <span class="st">&quot;Heme_LMPP&quot;</span>, <span class="st">&quot;Heme_Ery&quot;</span>, <span class="st">&quot;Heme_Mono&quot;</span>, <span class="st">&quot;Heme_CD4&quot;</span>, <span class="st">&quot;Heme_CD8&quot;</span>, <span class="st">&quot;Heme_B&quot;</span>, <span class="st">&quot;Heme_NK&quot;</span>, <span class="st">&quot;IAtlas_DC_Plasmacytoid&quot;</span>)</span>
<span id="cb265-2"><a href="archr-and-custom-deviations.html#cb265-2"></a><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(ATACPeaks, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;ATACMatrix&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “z:Heme_NK” “z:Heme_Mono”<br />
## [3] “z:Heme_LMPP” “z:Heme_HSC”<br />
## [5] “z:Heme_Ery” “z:Heme_CD8”<br />
## [7] “z:Heme_CD4” “z:Heme_B”<br />
## [9] “z:IAtlas_DC_Plasmacytoid” “deviations:Heme_NK”<br />
## [11] “deviations:Heme_Mono” “deviations:Heme_LMPP”<br />
## [13] “deviations:Heme_HSC” “deviations:Heme_Ery”<br />
## [15] “deviations:Heme_CD8” “deviations:Heme_CD4”<br />
## [17] “deviations:Heme_B” “deviations:IAtlas_DC_Plasmacytoid”</p>
</blockquote>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="archr-and-custom-deviations.html#cb266-1"></a>markerATAC &lt;-<span class="st"> </span><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(ATACPeaks, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;ATACMatrix&quot;</span>)</span>
<span id="cb266-2"><a href="archr-and-custom-deviations.html#cb266-2"></a>markerATAC &lt;-<span class="st"> </span><span class="kw">sort</span>(<span class="kw">grep</span>(<span class="st">&quot;z:&quot;</span>, markerATAC, <span class="dt">value =</span> <span class="ot">TRUE</span>))</span>
<span id="cb266-3"><a href="archr-and-custom-deviations.html#cb266-3"></a>markerATAC</span></code></pre></div>
<blockquote>
<p>## [1] “z:Heme_B” “z:Heme_CD4”<br />
## [3] “z:Heme_CD8” “z:Heme_Ery”<br />
## [5] “z:Heme_HSC” “z:Heme_LMPP”<br />
## [7] “z:Heme_Mono” “z:Heme_NK”<br />
## [9] “z:IAtlas_DC_Plasmacytoid”</p>
</blockquote>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="archr-and-custom-deviations.html#cb267-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb267-2"><a href="archr-and-custom-deviations.html#cb267-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb267-3"><a href="archr-and-custom-deviations.html#cb267-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;ATACMatrix&quot;</span>, </span>
<span id="cb267-4"><a href="archr-and-custom-deviations.html#cb267-4"></a>    <span class="dt">name =</span> markerATAC, </span>
<span id="cb267-5"><a href="archr-and-custom-deviations.html#cb267-5"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb267-6"><a href="archr-and-custom-deviations.html#cb267-6"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme5)</span>
<span id="cb267-7"><a href="archr-and-custom-deviations.html#cb267-7"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-1037416f3cb56-Date-2020-04-15_Time-11-13-23.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = ATACMatrix<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting Embedding<br />
## 1 2 3 4 5 6 7 8 9<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-1037416f3cb56-Date-2020-04-15_Time-11-13-23.log</p>
</blockquote>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="archr-and-custom-deviations.html#cb268-1"></a>p2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb268-2"><a href="archr-and-custom-deviations.html#cb268-2"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb268-3"><a href="archr-and-custom-deviations.html#cb268-3"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb268-4"><a href="archr-and-custom-deviations.html#cb268-4"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb268-5"><a href="archr-and-custom-deviations.html#cb268-5"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb268-6"><a href="archr-and-custom-deviations.html#cb268-6"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb268-7"><a href="archr-and-custom-deviations.html#cb268-7"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb268-8"><a href="archr-and-custom-deviations.html#cb268-8"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb268-9"><a href="archr-and-custom-deviations.html#cb268-9"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb268-10"><a href="archr-and-custom-deviations.html#cb268-10"></a>    )</span>
<span id="cb268-11"><a href="archr-and-custom-deviations.html#cb268-11"></a>})</span>
<span id="cb268-12"><a href="archr-and-custom-deviations.html#cb268-12"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-MarkerATAC-W-Imputation_1.png" width="800" height="800" /></p>
</div>
<div id="custom-deviations" class="section level3" number="13.2.3">
<h3><span class="header-section-number">13.2.3</span> Custom Deviations</h3>
<p>Instead of using the ArchR-curated region sets described above, we can provide our own custom region sets as a peak annotation. These custom annotations can be used in exactly the same way as the ArchR-curated annotations.</p>
<p>First, in case you haven’t already created this “EncodePeaks” annotation in a previous chapter, lets create it by downloading some ENCODE peak sets and calling <code>addPeakAnnotations()</code>.</p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="archr-and-custom-deviations.html#cb269-1"></a>EncodePeaks &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb269-2"><a href="archr-and-custom-deviations.html#cb269-2"></a>  <span class="dt">Encode_K562_GATA1 =</span> <span class="st">&quot;https://www.encodeproject.org/files/ENCFF632NQI/@@download/ENCFF632NQI.bed.gz&quot;</span>,</span>
<span id="cb269-3"><a href="archr-and-custom-deviations.html#cb269-3"></a>  <span class="dt">Encode_GM12878_CEBPB =</span> <span class="st">&quot;https://www.encodeproject.org/files/ENCFF761MGJ/@@download/ENCFF761MGJ.bed.gz&quot;</span>,</span>
<span id="cb269-4"><a href="archr-and-custom-deviations.html#cb269-4"></a>  <span class="dt">Encode_K562_Ebf1 =</span> <span class="st">&quot;https://www.encodeproject.org/files/ENCFF868VSY/@@download/ENCFF868VSY.bed.gz&quot;</span>,</span>
<span id="cb269-5"><a href="archr-and-custom-deviations.html#cb269-5"></a>  <span class="dt">Encode_K562_Pax5 =</span> <span class="st">&quot;https://www.encodeproject.org/files/ENCFF339KUO/@@download/ENCFF339KUO.bed.gz&quot;</span></span>
<span id="cb269-6"><a href="archr-and-custom-deviations.html#cb269-6"></a>)</span>
<span id="cb269-7"><a href="archr-and-custom-deviations.html#cb269-7"></a><span class="cf">if</span>(<span class="st">&quot;ChIP&quot;</span> <span class="op">%ni%</span><span class="st"> </span><span class="kw">names</span>(projHeme5<span class="op">@</span>peakAnnotation)){</span>
<span id="cb269-8"><a href="archr-and-custom-deviations.html#cb269-8"></a>    projHeme5 &lt;-<span class="st"> </span><span class="kw">addPeakAnnotations</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">regions =</span> EncodePeaks, <span class="dt">name =</span> <span class="st">&quot;ChIP&quot;</span>)</span>
<span id="cb269-9"><a href="archr-and-custom-deviations.html#cb269-9"></a>}</span></code></pre></div>
<p>Then, we make a deviations matrix from this peak annotation.</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="archr-and-custom-deviations.html#cb270-1"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addDeviationsMatrix</span>(</span>
<span id="cb270-2"><a href="archr-and-custom-deviations.html#cb270-2"></a>  <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb270-3"><a href="archr-and-custom-deviations.html#cb270-3"></a>  <span class="dt">peakAnnotation =</span> <span class="st">&quot;ChIP&quot;</span>,</span>
<span id="cb270-4"><a href="archr-and-custom-deviations.html#cb270-4"></a>  <span class="dt">force =</span> <span class="ot">TRUE</span></span>
<span id="cb270-5"><a href="archr-and-custom-deviations.html#cb270-5"></a>)</span></code></pre></div>
<blockquote>
<p>## Using Previous Background Peaks!<br />
## ArchR logging to : ArchRLogs/ArchR-addDeviationsMatrix-10374246a529b-Date-2020-04-15_Time-11-13-54.log<br />
## If there is an issue, please report to github with logFile!<br />
## NULL<br />
## 2020-04-15 11:13:59 : Batch Execution w/ safelapply!, 0 mins elapsed.<br />
## ###########<br />
## 2020-04-15 11:14:24 : Completed Computing Deviations!, 0.505 mins elapsed.<br />
## ###########<br />
## ArchR logging successful to : ArchRLogs/ArchR-addDeviationsMatrix-10374246a529b-Date-2020-04-15_Time-11-13-54.log</p>
</blockquote>
<p>The rest of the analysis workflow is the same as what has now been presented multiple times above.</p>
<p>We can plot the ranked deviations.</p>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="archr-and-custom-deviations.html#cb271-1"></a>plotVarDev &lt;-<span class="st"> </span><span class="kw">getVarDeviations</span>(projHeme5, <span class="dt">plot =</span> <span class="ot">TRUE</span>, <span class="dt">name =</span> <span class="st">&quot;ChIPMatrix&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## DataFrame with 4 rows and 6 columns<br />
## seqnames idx name combinedVars combinedMeans<br />
## <Rle> <array> <array> <numeric> <numeric><br />
## f1 z 1 Encode_K562_GATA1 6.33176528990239 -0.0165812177586936<br />
## f3 z 3 Encode_K562_Ebf1 3.41531511621115 0.0355188580388895<br />
## f4 z 4 Encode_K562_Pax5 3.24513143947069 -0.00960057929930531<br />
## f2 z 2 Encode_GM12878_CEBPB 0.822128491900276 0.00742820197777246<br />
## rank<br />
## <integer><br />
## f1 1<br />
## f3 2<br />
## f4 3<br />
## f2 4</p>
</blockquote>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="archr-and-custom-deviations.html#cb272-1"></a>plotVarDev</span></code></pre></div>
<blockquote>
<p>## Warning: Removed 21 rows containing missing values (geom_label_repel).</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Variable-ATAC-Deviation-Scores_1.png" width="500" height="500" /></p>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="archr-and-custom-deviations.html#cb273-1"></a><span class="kw">plotPDF</span>(plotVarDev, <span class="dt">name =</span> <span class="st">&quot;Variable-ChIP-Deviation-Scores&quot;</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## Warning: Removed 21 rows containing missing values (geom_label_repel).<br />
## Warning: Removed 21 rows containing missing values (geom_label_repel).<br />
## [1] 0</p>
</blockquote>
<p>And we can plot the deviation z-scores overlayed on our UMAP embedding.</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="archr-and-custom-deviations.html#cb274-1"></a>markerChIP &lt;-<span class="st"> </span><span class="kw">getFeatures</span>(projHeme5, <span class="dt">useMatrix =</span> <span class="st">&quot;ChIPMatrix&quot;</span>)</span>
<span id="cb274-2"><a href="archr-and-custom-deviations.html#cb274-2"></a>markerChIP &lt;-<span class="st"> </span><span class="kw">sort</span>(<span class="kw">grep</span>(<span class="st">&quot;z:&quot;</span>, markerChIP, <span class="dt">value =</span> <span class="ot">TRUE</span>))</span>
<span id="cb274-3"><a href="archr-and-custom-deviations.html#cb274-3"></a>markerChIP</span></code></pre></div>
<blockquote>
<p>## [1] “z:Encode_GM12878_CEBPB” “z:Encode_K562_Ebf1” “z:Encode_K562_GATA1”<br />
## [4] “z:Encode_K562_Pax5”</p>
</blockquote>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="archr-and-custom-deviations.html#cb275-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb275-2"><a href="archr-and-custom-deviations.html#cb275-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb275-3"><a href="archr-and-custom-deviations.html#cb275-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;ChIPMatrix&quot;</span>, </span>
<span id="cb275-4"><a href="archr-and-custom-deviations.html#cb275-4"></a>    <span class="dt">name =</span> markerChIP, </span>
<span id="cb275-5"><a href="archr-and-custom-deviations.html#cb275-5"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb275-6"><a href="archr-and-custom-deviations.html#cb275-6"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme5)</span>
<span id="cb275-7"><a href="archr-and-custom-deviations.html#cb275-7"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-10374635d5970-Date-2020-04-15_Time-11-14-35.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = ChIPMatrix<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting Embedding<br />
## 1 2 3 4<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-10374635d5970-Date-2020-04-15_Time-11-14-35.log</p>
</blockquote>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="archr-and-custom-deviations.html#cb276-1"></a>p2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb276-2"><a href="archr-and-custom-deviations.html#cb276-2"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb276-3"><a href="archr-and-custom-deviations.html#cb276-3"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb276-4"><a href="archr-and-custom-deviations.html#cb276-4"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb276-5"><a href="archr-and-custom-deviations.html#cb276-5"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb276-6"><a href="archr-and-custom-deviations.html#cb276-6"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb276-7"><a href="archr-and-custom-deviations.html#cb276-7"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb276-8"><a href="archr-and-custom-deviations.html#cb276-8"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb276-9"><a href="archr-and-custom-deviations.html#cb276-9"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb276-10"><a href="archr-and-custom-deviations.html#cb276-10"></a>    )</span>
<span id="cb276-11"><a href="archr-and-custom-deviations.html#cb276-11"></a>})</span>
<span id="cb276-12"><a href="archr-and-custom-deviations.html#cb276-12"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">2</span>),p2))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-MarkerChIP-W-Imputation_1.png" width="800" height="800" /></p>

</div>
</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="motif-deviations.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="footprinting-with-archr.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/13_Deviations.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/13_Deviations.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
},
"search": true,
"info": false
});
});
</script>

</body>

</html>
