<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8.1 Cross-platform linkage of scATAC-seq cells with scRNA-seq cells | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="8.1 Cross-platform linkage of scATAC-seq cells with scRNA-seq cells | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8.1 Cross-platform linkage of scATAC-seq cells with scRNA-seq cells | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-05-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="defining-cluster-identity-with-scrna-seq.html"/>
<link rel="next" href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://www.archrproject.com/index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li><a href="index.html#section"><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>1.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="1.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>1.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="1.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>1.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>1.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="1.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="1.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>1.7</b> Per-cell Quality Control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>2.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="2.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="3.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>4.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="4.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>4.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="4.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>5.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>6.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="6.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>6.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="6.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>6.3</b> Dimensionality Reduction After Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>7</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>7.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="7.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>7.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="7.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>7.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="7.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>7.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="7.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="7.6" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.6</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.7" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.7</b> Launching the ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>8</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>8.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>9</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="9.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>10.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="10.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>10.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="10.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>10.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="10.4" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>10.4</b> Add Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>11</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>11.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="11.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>11.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="11.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>11.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-enrichment-in-differential-peaks.html"><a href="motif-enrichment-in-differential-peaks.html"><i class="fa fa-check"></i><b>12.1</b> Motif Enrichment in Differential Peaks</a></li>
<li class="chapter" data-level="12.2" data-path="motif-enrichment-in-marker-peaks.html"><a href="motif-enrichment-in-marker-peaks.html"><i class="fa fa-check"></i><b>12.2</b> Motif Enrichment in Marker Peaks</a></li>
<li class="chapter" data-level="12.3" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>12.3</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="12.4" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>12.4</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>13</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>13.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="13.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>13.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>14</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>14.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="14.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>14.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="14.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>14.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>15</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>15.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="15.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>15.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="15.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>15.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="15.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>15.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>16</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="myeloid-trajectory-monocyte-differentiation.html"><a href="myeloid-trajectory-monocyte-differentiation.html"><i class="fa fa-check"></i><b>16.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="16.2" data-path="lymphoid-trajectory-b-cell-cifferentiation.html"><a href="lymphoid-trajectory-b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>16.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>17</b> Plot aesthetics in ArchR</a></li>
<li class="chapter" data-level="18" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>18</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="index.html">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</h2>
<p>In order to integrate our tutorial scATAC-seq data with matched scRNA-seq data, we will use scRNA-seq data derived from the same hematopoietic cell types from Granja* et al (2019).</p>
<p>We have stored this scRNA-seq data as a 111 MB <code>RangedSummarizedExperiment</code> object. However, ArchR also accepts unmodified <code>Seurat</code> objects as input to the integration workflow. Downloading and examining this object, we see that it has a gene expression counts matrix and associated metadata.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb114-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(<span class="st">&quot;scRNA-Hematopoiesis-Granja-2019.rds&quot;</span>)){</span>
<span id="cb114-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb114-2"></a>    <span class="kw">download.file</span>(</span>
<span id="cb114-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb114-3"></a>        <span class="dt">url =</span> <span class="st">&quot;https://jeffgranja.s3.amazonaws.com/ArchR/TestData/scRNA-Hematopoiesis-Granja-2019.rds&quot;</span>,</span>
<span id="cb114-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb114-4"></a>        <span class="dt">destfile =</span> <span class="st">&quot;scRNA-Hematopoiesis-Granja-2019.rds&quot;</span></span>
<span id="cb114-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb114-5"></a>    )</span>
<span id="cb114-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb114-6"></a>}</span>
<span id="cb114-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb114-7"></a></span>
<span id="cb114-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb114-8"></a>seRNA &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;scRNA-Hematopoiesis-Granja-2019.rds&quot;</span>)</span>
<span id="cb114-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb114-9"></a>seRNA</span></code></pre></div>
<blockquote>
<p>## class: RangedSummarizedExperiment<br />
## dim: 20287 35582<br />
## metadata(0):<br />
## assays(1): counts<br />
## rownames(20287): FAM138A OR4F5 … S100B PRMT2<br />
## rowData names(3): gene_name gene_id exonLength<br />
## colnames(35582): CD34_32_R5:AAACCTGAGTATCGAA-1<br />
## CD34_32_R5:AAACCTGAGTCGTTTG-1 …<br />
## BMMC_10x_GREENLEAF_REP2:TTTGTTGCATGTGTCA-1<br />
## BMMC_10x_GREENLEAF_REP2:TTTGTTGCATTGAAAG-1<br />
## colData names(10): Group nUMI_pre … BioClassification Barcode</p>
</blockquote>
<p>The metadata contains a column called <code>BioClassification</code> which contains the cell type classifications for each cell in the scRNA-seq dataset.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb115-1"></a><span class="kw">colnames</span>(<span class="kw">colData</span>(seRNA))</span></code></pre></div>
<blockquote>
<p>## [1] “Group” “nUMI_pre” “nUMI”<br />
## [4] “nGene” “initialClusters” “UMAP1”<br />
## [7] “UMAP2” “Clusters” “BioClassification”<br />
## [10] “Barcode”</p>
</blockquote>
<p>Using <code>table()</code> we can see how many cells are in each of the scRNA-seq cell type classifications.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb116-1"></a><span class="kw">table</span>(<span class="kw">colData</span>(seRNA)<span class="op">$</span>BioClassification)</span></code></pre></div>
<blockquote>
<p>## 01_HSC 02_Early.Eryth 03_Late.Eryth 04_Early.Baso 05_CMP.LMPP<br />
## 1425 1653 446 111 2260<br />
## 06_CLP.1 07_GMP 08_GMP.Neut 09_pDC 10_cDC<br />
## 903 2097 1050 544 325<br />
## 11_CD14.Mono.1 12_CD14.Mono.2 13_CD16.Mono 14_Unk 15_CLP.2<br />
## 1800 4222 292 520 377<br />
## 16_Pre.B 17_B 18_Plasma 19_CD8.N 20_CD4.N1<br />
## 710 1711 62 1521 2470<br />
## 21_CD4.N2 22_CD4.M 23_CD8.EM 24_CD8.CM 25_NK<br />
## 2364 3539 796 2080 2143<br />
## 26_Unk<br />
## 161</p>
</blockquote>
<p>There are two types of integration approaches that we can perform. <strong>Unconstrained integration</strong> is a completely agnostic approach that would take all of the cells in your scATAC-seq experiment and attempt to align them to any of the cells in the scRNA-seq experiment. While this is a feasible preliminary solution, we can improve the quality of our cross-platform alignment by <em>constraining</em> the integration process. To perform a <strong>constrained integration</strong> we use prior knowledge of the cell types to limit the search space of the alignment. For example, if we knew that Clusters A, B, and C in the scATAC-seq data corresponded to 3 different T cell clusters, and we knew that Clusters X and Y in the scRNA-seq data corresponded to 2 different T cell clusters, we could tell ArchR to specifically try to align cells from scATAC-seq Clusters A, B, and C to cell from scRNA-seq clusters X and Y. We illustrate these two approaches below, first performing an unconstrained integration to acheive preliminary cluster identities and then using this prior knowledge to perform a more refined constrained integration.</p>
<div id="unconstrained-integration" class="section level3" number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> Unconstrained Integration</h3>
<p>To integrate scATAC-seq with scRNA-seq, we use the <code>addGeneIntegrationMatrix()</code> function. As mentioned above, this function accepts either a <code>Seurat</code> object or a <code>RangedSummarizedExperiment</code> object via the <code>seRNA</code> parameter. The first round of integration we perform will be a preliminary <strong>unconstrained</strong> integration and we will not store this in the Arrow files (<code>addToArrow = FALSE</code>). We provide a name for the integration matrix that will be stored in the <code>ArchRProject</code> via the <code>matrixName</code> parameter. The other key parameters for this function provide column names in <code>cellColData</code> where certain information will be stored: <code>nameCell</code> will store the cell ID from the matched scRNA-seq cell, <code>nameGroup</code> will store the group ID from the scRNA-seq cell, and <code>nameScore</code> will store the cross-platform integration score.</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-1"></a>projHeme2 &lt;-<span class="st"> </span><span class="kw">addGeneIntegrationMatrix</span>(</span>
<span id="cb117-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-2"></a>    <span class="dt">ArchRProj =</span> projHeme2, </span>
<span id="cb117-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-3"></a>    <span class="dt">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>,</span>
<span id="cb117-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-4"></a>    <span class="dt">matrixName =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>,</span>
<span id="cb117-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-5"></a>    <span class="dt">reducedDims =</span> <span class="st">&quot;IterativeLSI&quot;</span>,</span>
<span id="cb117-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-6"></a>    <span class="dt">seRNA =</span> seRNA,</span>
<span id="cb117-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-7"></a>    <span class="dt">addToArrow =</span> <span class="ot">FALSE</span>,</span>
<span id="cb117-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-8"></a>    <span class="dt">groupRNA =</span> <span class="st">&quot;BioClassification&quot;</span>,</span>
<span id="cb117-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-9"></a>    <span class="dt">nameCell =</span> <span class="st">&quot;predictedCell_Un&quot;</span>,</span>
<span id="cb117-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-10"></a>    <span class="dt">nameGroup =</span> <span class="st">&quot;predictedGroup_Un&quot;</span>,</span>
<span id="cb117-11"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-11"></a>    <span class="dt">nameScore =</span> <span class="st">&quot;predictedScore_Un&quot;</span></span>
<span id="cb117-12"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb117-12"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-addGeneIntegrationMatrix-f6635e453cfd-Date-2020-04-15_Time-10-08-55.log<br />
## If there is an issue, please report to github with logFile!<br />
## 2020-04-15 10:08:56 : Running Seurat’s Integration Stuart* et al 2019, 0.005 mins elapsed.<br />
## 2020-04-15 10:09:06 : Checking ATAC Input, 0.172 mins elapsed.<br />
## 2020-04-15 10:09:06 : Checking RNA Input, 0.173 mins elapsed.<br />
## 2020-04-15 10:09:17 : Creating Integration Blocks, 0.367 mins elapsed.<br />
## 2020-04-15 10:09:18 : Prepping Interation Data, 0.372 mins elapsed.<br />
## 2020-04-15 10:09:19 : Computing Integration in 1 Integration Blocks!, 0 mins elapsed.<br />
## 2020-04-15 10:09:19 : Block (1 of 1) : Computing Integration, 0 mins elapsed.<br />
## 2020-04-15 10:09:23 : Block (1 of 1) : Identifying Variable Genes, 0.071 mins elapsed.<br />
## 2020-04-15 10:09:28 : Block (1 of 1) : Getting GeneScoreMatrix, 0.146 mins elapsed.<br />
## 2020-04-15 10:09:37 : Block (1 of 1) : Imputing GeneScoreMatrix, 0.298 mins elapsed.<br />
## 2020-04-15 10:09:37 : Computing Impute Weights Using Magic (Cell 2018), 0 mins elapsed.<br />
## 2020-04-15 10:09:37 : Computing Partial Diffusion Matrix with Magic (1 of 2), 0 mins elapsed.<br />
## 2020-04-15 10:09:44 : Computing Partial Diffusion Matrix with Magic (2 of 2), 0.122 mins elapsed.<br />
## 2020-04-15 10:09:51 : Completed Getting Magic Weights!, 0.242 mins elapsed.<br />
## Getting ImputeWeights<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Getting ImputeWeights<br />
## 2020-04-15 10:10:17 : Block (1 of 1) : Seurat FindTransferAnchors, 0.968 mins elapsed.<br />
## 2020-04-15 10:12:00 : Block (1 of 1) : Seurat TransferData Cell Labels, 2.681 mins elapsed.<br />
## 2020-04-15 10:12:34 : Block (1 of 1) : Completed Integration, 3.243 mins elapsed.<br />
## 2020-04-15 10:12:34 : Completed Integration with RNA Matrix, 3.256 mins elapsed.<br />
## ArchR logging successful to : ArchRLogs/ArchR-addGeneIntegrationMatrix-f6635e453cfd-Date-2020-04-15_Time-10-08-55.log</p>
</blockquote>
<p>This unconstrained integration can be used as shown below to perform a more refined constrained integration.</p>
</div>
<div id="constrained-integration" class="section level3" number="8.1.2">
<h3><span class="header-section-number">8.1.2</span> Constrained Integration</h3>
<p>Now that we have our preliminary unconstrained integration, we will identify general cell types to profide a framework to further refine the integration results.</p>
<p>Given that this tutorial data is from hematopoietic cells, we would ideally constrain the integration to associate similar cell types togther. First, we will identify which cell types from the scRNA-seq data are most abundant in each of our scATAC-seq clusters. The goal of this is to identify the cells in both the scATAC-seq and scRNA-seq data that correspond to T cells and NK cells using the unconstrained integration so that we can perform a constrained integration using this information. To do this, we will create a <code>confusionMatrix</code> that looks at the intersection of <code>Clusters</code> and <code>predictedGroup_Un</code> which contains the cell types as identified by scRNA-seq.</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb118-1"></a>cM &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">confusionMatrix</span>(projHeme2<span class="op">$</span>Clusters, projHeme2<span class="op">$</span>predictedGroup_Un))</span>
<span id="cb118-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb118-2"></a>preClust &lt;-<span class="st"> </span><span class="kw">colnames</span>(cM)[<span class="kw">apply</span>(cM, <span class="dv">1</span> , which.max)]</span>
<span id="cb118-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb118-3"></a><span class="kw">cbind</span>(preClust, <span class="kw">rownames</span>(cM)) <span class="co">#Assignments</span></span></code></pre></div>
<blockquote>
<p>## preClust<br />
## [1,] “17_B” “C3”<br />
## [2,] “20_CD4.N1” “C8”<br />
## [3,] “16_Pre.B” “C4”<br />
## [4,] “08_GMP.Neut” “C11”<br />
## [5,] “11_CD14.Mono.1” “C1”<br />
## [6,] “01_HSC” “C12”<br />
## [7,] “03_Late.Eryth” “C10”<br />
## [8,] “22_CD4.M” “C9”<br />
## [9,] “25_NK” “C7”<br />
## [10,] “09_pDC” “C5”<br />
## [11,] “12_CD14.Mono.2” “C2”<br />
## [12,] “15_CLP.2” “C6”</p>
</blockquote>
<p>The above list shows which scRNA-seq cell type is most abundant in each of the 12 scATAC-seq clusters.</p>
<p>First, lets look at the cell type labels from our scRNA-seq data that were used in our unconstrained integration:</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb119-1"></a><span class="kw">unique</span>(<span class="kw">unique</span>(projHeme2<span class="op">$</span>predictedGroup_Un))</span></code></pre></div>
<blockquote>
<p>## [1] “05_CMP.LMPP” “08_GMP.Neut” “01_HSC” “06_CLP.1”
## [5] “15_CLP.2” “02_Early.Eryth” “07_GMP” “09_pDC”
## [9] “04_Early.Baso” “03_Late.Eryth” “17_B” “12_CD14.Mono.2”
## [13] “16_Pre.B” “10_cDC” “11_CD14.Mono.1” “25_NK”
## [17] “21_CD4.N2” “22_CD4.M” “23_CD8.EM” “19_CD8.N”
## [21] “24_CD8.CM” “26_Unk” “20_CD4.N1” “14_Unk”
## [25] “13_CD16.Mono” “18_Plasma”</p>
</blockquote>
<p>From the above list, we can see that the clusters in the scRNA-seq data that correspond to T cells and NK cells are Clusters 19-25. We will create a string-based representation of these clusters to use in the downstream constrained integration.</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb120-1"></a><span class="co">#From scRNA</span></span>
<span id="cb120-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb120-2"></a>cTNK &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">paste0</span>(<span class="dv">19</span><span class="op">:</span><span class="dv">25</span>), <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>)</span>
<span id="cb120-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb120-3"></a>cTNK</span></code></pre></div>
<blockquote>
<p>## [1] “19|20|21|22|23|24|25”</p>
</blockquote>
<p>We can then take all of the other clusters and create a string-based representation of all “Non-T cell, Non-NK cell” clusters (i.e. Cluster 1 - 18).</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb121-1"></a>cNonTNK &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="kw">paste0</span>(<span class="st">&quot;0&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">9</span>), <span class="dv">10</span><span class="op">:</span><span class="dv">13</span>, <span class="dv">15</span><span class="op">:</span><span class="dv">18</span>), <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>)</span>
<span id="cb121-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb121-2"></a>cNonTNK</span></code></pre></div>
<blockquote>
<p>## [1] “01|02|03|04|05|06|07|08|09|10|11|12|13|15|16|17|18”</p>
</blockquote>
<p>These string-based representations are pattern matching strings that we will use with <code>grep</code> to extract the scATAC-seq clusters that correspond to these scRNA-seq cell types. The <code>|</code> in the string acts as an <code>or</code> statement so we end up searching for any row in the <code>preClust</code> column of our confusion matrix that matches one of the scRNA-seq cluster numbers provided in the pattern match string.</p>
<p>For T cells and NK cells, this identifies scATAC-seq clusters C7, C8, and C9:</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb122-1"></a><span class="co">#Assign scATAC to these categories</span></span>
<span id="cb122-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb122-2"></a>clustTNK &lt;-<span class="st"> </span><span class="kw">rownames</span>(cM)[<span class="kw">grep</span>(cTNK, preClust)]</span>
<span id="cb122-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb122-3"></a>clustTNK</span></code></pre></div>
<blockquote>
<p>## [1] “C8” “C9” “C7”</p>
</blockquote>
<p>For Non-T cells and Non-NK cells, this identifies the remaining scATAC-seq clusters:</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb123-1"></a>clustNonTNK &lt;-<span class="st"> </span><span class="kw">rownames</span>(cM)[<span class="kw">grep</span>(cNonTNK, preClust)]</span>
<span id="cb123-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb123-2"></a>clustNonTNK</span></code></pre></div>
<blockquote>
<p>## [1] “C3” “C4” “C11” “C1” “C12” “C10” “C5” “C2” “C6”</p>
</blockquote>
<p>We then perform a similar opperation to identify the scRNA-seq cells that correspond to these same cell types.
First, we identify the T cell and NK cells in the scRNA-seq data</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb124-1"></a><span class="co">#RNA get cells in these categories</span></span>
<span id="cb124-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb124-2"></a>rnaTNK &lt;-<span class="st"> </span><span class="kw">colnames</span>(seRNA)[<span class="kw">grep</span>(cTNK, <span class="kw">colData</span>(seRNA)<span class="op">$</span>BioClassification)]</span>
<span id="cb124-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb124-3"></a><span class="kw">head</span>(rnaTNK)</span></code></pre></div>
<blockquote>
<p>## [1] “PBMC_10x_GREENLEAF_REP1:AAACCCAGTCGTCATA-1”<br />
## [2] “PBMC_10x_GREENLEAF_REP1:AAACCCATCCGATGTA-1”<br />
## [3] “PBMC_10x_GREENLEAF_REP1:AAACCCATCTCAACGA-1”<br />
## [4] “PBMC_10x_GREENLEAF_REP1:AAACCCATCTCTCGAC-1”<br />
## [5] “PBMC_10x_GREENLEAF_REP1:AAACGAACAATCGTCA-1”<br />
## [6] “PBMC_10x_GREENLEAF_REP1:AAACGAACACGATTCA-1”</p>
</blockquote>
<p>Then, we identify the Non-T cell Non-NK cell cells in the scRNA-seq data.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb125-1"></a>rnaNonTNK &lt;-<span class="st"> </span><span class="kw">colnames</span>(seRNA)[<span class="kw">grep</span>(cNonTNK, <span class="kw">colData</span>(seRNA)<span class="op">$</span>BioClassification)]</span>
<span id="cb125-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb125-2"></a><span class="kw">head</span>(rnaNonTNK)</span></code></pre></div>
<blockquote>
<p>## [1] “CD34_32_R5:AAACCTGAGTATCGAA-1” “CD34_32_R5:AAACCTGAGTCGTTTG-1”<br />
## [3] “CD34_32_R5:AAACCTGGTTCCACAA-1” “CD34_32_R5:AAACGGGAGCTTCGCG-1”<br />
## [5] “CD34_32_R5:AAACGGGAGGGAGTAA-1” “CD34_32_R5:AAACGGGAGTTACGGG-1”</p>
</blockquote>
<p>To prepare for this constrained integration, we create a nested list. This is a <code>SimpleList</code> of multiple <code>SimpleList</code> objects, one for each group that we would like to constrain. In this example, we have two groups: one group called <code>TNK</code> that identifies the T cells and NK cells across the two platforms, and a second group called <code>NonTNK</code> that identifies the Non-T cell Non-NK cell cells across the two platforms. Each of these <code>SimpleList</code> objects has two vectors of cell IDs, one called <code>ATAC</code> and one called <code>RNA</code> as shown below:</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-1"></a>groupList &lt;-<span class="st"> </span><span class="kw">SimpleList</span>(</span>
<span id="cb126-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-2"></a>    <span class="dt">TNK =</span> <span class="kw">SimpleList</span>(</span>
<span id="cb126-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-3"></a>        <span class="dt">ATAC =</span> projHeme2<span class="op">$</span>cellNames[projHeme2<span class="op">$</span>Clusters <span class="op">%in%</span><span class="st"> </span>clustTNK],</span>
<span id="cb126-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-4"></a>        <span class="dt">RNA =</span> rnaTNK</span>
<span id="cb126-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-5"></a>    ),</span>
<span id="cb126-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-6"></a>    <span class="dt">NonTNK =</span> <span class="kw">SimpleList</span>(</span>
<span id="cb126-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-7"></a>        <span class="dt">ATAC =</span> projHeme2<span class="op">$</span>cellNames[projHeme2<span class="op">$</span>Clusters <span class="op">%in%</span><span class="st"> </span>clustNonTNK],</span>
<span id="cb126-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-8"></a>        <span class="dt">RNA =</span> rnaNonTNK</span>
<span id="cb126-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-9"></a>    )    </span>
<span id="cb126-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-10"></a>)</span>
<span id="cb126-11"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-11"></a></span>
<span id="cb126-12"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-12"></a>We pass this list to the <span class="st">`</span><span class="dt">groupList</span><span class="st">`</span> parameter of the <span class="st">`</span><span class="dt">addGeneIntegrationMatrix()</span><span class="st">`</span> <span class="cf">function</span> to constrain our integration. Note that, <span class="cf">in</span> this case, we are still not adding these results to the Arrow <span class="kw">files</span> (<span class="st">`</span><span class="dt">addToArrow = FALSE</span><span class="st">`</span>). We recommend checking the results of the integration thoroughly against your expectations prior to saving the results <span class="cf">in</span> the Arrow files. We illustrate this process <span class="cf">in</span> the <span class="cf">next</span> section of the book.</span>
<span id="cb126-13"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-13"></a><span class="co">#~5 minutes</span></span>
<span id="cb126-14"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-14"></a>projHeme2 &lt;-<span class="st"> </span><span class="kw">addGeneIntegrationMatrix</span>(</span>
<span id="cb126-15"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-15"></a>    <span class="dt">ArchRProj =</span> projHeme2, </span>
<span id="cb126-16"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-16"></a>    <span class="dt">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>,</span>
<span id="cb126-17"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-17"></a>    <span class="dt">matrixName =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>,</span>
<span id="cb126-18"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-18"></a>    <span class="dt">reducedDims =</span> <span class="st">&quot;IterativeLSI&quot;</span>,</span>
<span id="cb126-19"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-19"></a>    <span class="dt">seRNA =</span> seRNA,</span>
<span id="cb126-20"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-20"></a>    <span class="dt">addToArrow =</span> <span class="ot">FALSE</span>, </span>
<span id="cb126-21"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-21"></a>    <span class="dt">groupList =</span> groupList,</span>
<span id="cb126-22"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-22"></a>    <span class="dt">groupRNA =</span> <span class="st">&quot;BioClassification&quot;</span>,</span>
<span id="cb126-23"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-23"></a>    <span class="dt">nameCell =</span> <span class="st">&quot;predictedCell_Co&quot;</span>,</span>
<span id="cb126-24"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-24"></a>    <span class="dt">nameGroup =</span> <span class="st">&quot;predictedGroup_Co&quot;</span>,</span>
<span id="cb126-25"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-25"></a>    <span class="dt">nameScore =</span> <span class="st">&quot;predictedScore_Co&quot;</span></span>
<span id="cb126-26"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb126-26"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-addGeneIntegrationMatrix-f663618b2629-Date-2020-04-15_Time-10-12-35.log<br />
## If there is an issue, please report to github with logFile!<br />
## 2020-04-15 10:12:36 : Running Seurat’s Integration Stuart* et al 2019, 0.014 mins elapsed.<br />
## 2020-04-15 10:12:36 : Checking ATAC Input, 0.027 mins elapsed.<br />
## 2020-04-15 10:12:36 : Checking RNA Input, 0.027 mins elapsed.<br />
## 2020-04-15 10:12:48 : Creating Integration Blocks, 0.218 mins elapsed.<br />
## 2020-04-15 10:12:48 : Prepping Interation Data, 0.222 mins elapsed.<br />
## 2020-04-15 10:12:49 : Computing Integration in 2 Integration Blocks!, 0 mins elapsed.<br />
## 2020-04-15 10:15:24 : Completed Integration with RNA Matrix, 2.574 mins elapsed.<br />
## ArchR logging successful to : ArchRLogs/ArchR-addGeneIntegrationMatrix-f663618b2629-Date-2020-04-15_Time-10-12-35.log</p>
</blockquote>
</div>
<div id="comparing-unconstrained-and-constrained-integrations" class="section level3" number="8.1.3">
<h3><span class="header-section-number">8.1.3</span> Comparing Unconstrained and Constrained Integrations</h3>
<p>To compare our unconstrained and constrained integrations, we will color the cells in our scATAC-seq data based on the scRNA-seq cell type identified through integration. To do this, we will create a color palette using the built-in ArchR function <code>paletteDiscrete()</code>.</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb127-1"></a>pal &lt;-<span class="st"> </span><span class="kw">paletteDiscrete</span>(<span class="dt">values =</span> <span class="kw">colData</span>(seRNA)<span class="op">$</span>BioClassification)</span></code></pre></div>
<blockquote>
<p>## Length of unique values greater than palette, interpolating..</p>
</blockquote>
<p>In ArchR, a palette is essentially a named vector where the values are hex codes corresponding to the color to be associated with the names.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb128-1"></a>pal</span></code></pre></div>
<blockquote>
<p>## 01_HSC 02_Early.Eryth 03_Late.Eryth 04_Early.Baso 05_CMP.LMPP<br />
## “#D51F26” “#502A59” “#235D55” “#3D6E57” “#8D2B8B”<br />
## 06_CLP.1 07_GMP 08_GMP.Neut 09_pDC 10_cDC<br />
## “#DE6C3E” “#F9B712” “#D8CE42” “#8E9ACD” “#B774B1”<br />
## 11_CD14.Mono.1 12_CD14.Mono.2 13_CD16.Mono 14_Unk 15_CLP.2<br />
## “#D69FC8” “#C7C8DE” “#8FD3D4” “#89C86E” “#CC9672”<br />
## 16_Pre.B 17_B 18_Plasma 19_CD8.N 20_CD4.N1<br />
## “#CF7E96” “#A27AA4” “#CD4F32” “#6B977E” “#518AA3”<br />
## 21_CD4.N2 22_CD4.M 23_CD8.EM 24_CD8.CM 25_NK<br />
## “#5A5297” “#0F707D” “#5E2E32” “#A95A3C” “#B28D5C”<br />
## 26_Unk<br />
## “#3D3D3D”</p>
</blockquote>
<p>We can now visualize the integration by overlaying the scRNA-seq cell types on our scATAC-seq data based on the unconstrained integration.</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb129-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb129-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb129-2"></a>    projHeme2, </span>
<span id="cb129-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb129-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, </span>
<span id="cb129-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb129-4"></a>    <span class="dt">name =</span> <span class="st">&quot;predictedGroup_Un&quot;</span>, </span>
<span id="cb129-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb129-5"></a>    <span class="dt">pal =</span> pal</span>
<span id="cb129-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb129-6"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-f66351b51d5d-Date-2020-04-15_Time-10-15-31.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = cellColData<br />
## Plotting Embedding<br />
## 1<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-f66351b51d5d-Date-2020-04-15_Time-10-15-31.log</p>
</blockquote>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb130-1"></a>p1</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-RNA-Integration_1.png" width="600" height="600" /></p>
<p>Similarly, we can visualize the integration by overlaying the scRNA-seq cell types on our scATAC-seq data based on the constrained integration.</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb131-1"></a>p2 &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb131-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb131-2"></a>    projHeme2, </span>
<span id="cb131-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb131-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, </span>
<span id="cb131-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb131-4"></a>    <span class="dt">name =</span> <span class="st">&quot;predictedGroup_Co&quot;</span>, </span>
<span id="cb131-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb131-5"></a>    <span class="dt">pal =</span> pal</span>
<span id="cb131-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb131-6"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-f6632ef266a4-Date-2020-04-15_Time-10-15-49.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = cellColData<br />
## Plotting Embedding<br />
## 1<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-f6632ef266a4-Date-2020-04-15_Time-10-15-49.log</p>
</blockquote>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb132-1"></a>p2</span></code></pre></div>
<p>The differences between these the unconstrained and constrained integration is very subtle in this example, largely because the cell types of interest are already very distinct. However, you should notice differences, especially in the T cell clusters (Clusters 17-22).</p>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-RNA-Integration_2.png" width="600" height="600" /></p>
<p>To save an editable vectorized version of this plot, we use the <code>plotPDF()</code> function.</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb133-1"></a><span class="kw">plotPDF</span>(p1,p2, <span class="dt">name =</span> <span class="st">&quot;Plot-UMAP-RNA-Integration.pdf&quot;</span>, <span class="dt">ArchRProj =</span> projHeme2, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## [1] “plotting ggplot!”<br />
## [1] 0</p>
</blockquote>
<p>We can now save our <code>projHeme2</code> using the <code>saveArchRProject()</code> function.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb134-1"></a><span class="kw">saveArchRProject</span>(<span class="dt">ArchRProj =</span> projHeme2, <span class="dt">outputDirectory =</span> <span class="st">&quot;Save-ProjHeme2&quot;</span>, <span class="dt">load =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## Copying ArchRProject to new outputDirectory : /oak/stanford/groups/howchang/users/jgranja/ArchRTutorial/ArchRBook/BookOutput4/Save-ProjHeme2<br />
## Copying Arrow Files…<br />
## Copying Arrow Files (1 of 3)<br />
## Copying Arrow Files (2 of 3)<br />
## Copying Arrow Files (3 of 3)<br />
## Getting ImputeWeights<br />
## Dropping ImputeWeights…<br />
## Copying Other Files…<br />
## Copying Other Files (1 of 4): Embeddings<br />
## Copying Other Files (2 of 4): IterativeLSI<br />
## Copying Other Files (3 of 4): IterativeLSI2<br />
## Copying Other Files (4 of 4): Plots<br />
## Saving ArchRProject…</p>
</blockquote>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="defining-cluster-identity-with-scrna-seq.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/08_scRNA_Labeling.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/08_scRNA_Labeling.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
},
"search": true,
"info": false
});
});
</script>

</body>

</html>
