<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.2 Manipulating An ArchRProject | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="3.2 Manipulating An ArchRProject | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.2 Manipulating An ArchRProject | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-05-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="creating-an-archrproject-1.html"/>
<link rel="next" href="plotting-sample-statistics-from-an-archrproject.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://www.archrproject.com/index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li><a href="index.html#section"><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>1.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="1.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>1.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="1.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>1.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>1.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="1.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="1.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>1.7</b> Per-cell Quality Control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>2.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="2.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="3.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>4.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="4.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>4.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="4.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>5.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>6.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="6.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>6.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="6.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>6.3</b> Dimensionality Reduction After Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>7</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>7.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="7.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>7.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="7.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>7.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="7.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>7.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="7.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="7.6" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.6</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.7" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.7</b> Launching the ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>8</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>8.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>9</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="9.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>10.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="10.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>10.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="10.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>10.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="10.4" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>10.4</b> Add Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>11</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>11.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="11.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>11.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="11.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>11.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-enrichment-in-differential-peaks.html"><a href="motif-enrichment-in-differential-peaks.html"><i class="fa fa-check"></i><b>12.1</b> Motif Enrichment in Differential Peaks</a></li>
<li class="chapter" data-level="12.2" data-path="motif-enrichment-in-marker-peaks.html"><a href="motif-enrichment-in-marker-peaks.html"><i class="fa fa-check"></i><b>12.2</b> Motif Enrichment in Marker Peaks</a></li>
<li class="chapter" data-level="12.3" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>12.3</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="12.4" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>12.4</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>13</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>13.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="13.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>13.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>14</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>14.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="14.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>14.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="14.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>14.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>15</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>15.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="15.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>15.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="15.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>15.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="15.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>15.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>16</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="myeloid-trajectory-monocyte-differentiation.html"><a href="myeloid-trajectory-monocyte-differentiation.html"><i class="fa fa-check"></i><b>16.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="16.2" data-path="lymphoid-trajectory-b-cell-cifferentiation.html"><a href="lymphoid-trajectory-b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>16.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>17</b> Plot aesthetics in ArchR</a></li>
<li class="chapter" data-level="18" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>18</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="index.html">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="manipulating-an-archrproject" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Manipulating An ArchRProject</h2>
<p>Now that we have created an <code>ArchRProject</code>, there are many things that we can do to easily access or manipulate the associated data.</p>
<p><span style="font-size:20px;font-weight:bold">Example 1. The <code>$</code> accessor allows direct access to <code>cellColData</code></span></p>
<p>We can access the cell names associated with each cell:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="manipulating-an-archrproject.html#cb26-1"></a><span class="kw">head</span>(projHeme1<span class="op">$</span>cellNames)</span></code></pre></div>
<blockquote>
<p>## [1] “scATAC_BMMC_R1#TTATGTCAGTGATTAG-1” “scATAC_BMMC_R1#AAGATAGTCACCGCGA-1”<br />
## [3] “scATAC_BMMC_R1#GCATTGAAGATTCCGT-1” “scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1”<br />
## [5] “scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1” “scATAC_BMMC_R1#AGTTACGAGAACGTCG-1”</p>
</blockquote>
<p>We can access the sample names associated with each cell:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="manipulating-an-archrproject.html#cb27-1"></a><span class="kw">head</span>(projHeme1<span class="op">$</span>Sample)</span></code></pre></div>
<blockquote>
<p>## [1] “scATAC_BMMC_R1” “scATAC_BMMC_R1” “scATAC_BMMC_R1” “scATAC_BMMC_R1”<br />
## [5] “scATAC_BMMC_R1” “scATAC_BMMC_R1”</p>
</blockquote>
<p>We can access the TSS Enrichment Scores for each cell:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="manipulating-an-archrproject.html#cb28-1"></a><span class="kw">quantile</span>(projHeme1<span class="op">$</span>TSSEnrichment)</span></code></pre></div>
<blockquote>
<p>## 0% 25% 50% 75% 100%<br />
## 4.027 13.922 16.832 19.937 41.782</p>
</blockquote>
<p><span style="font-size:20px;font-weight:bold">Example 2. Subsetting an <code>ArchRProject</code> by cells</span></p>
<p>There are many ways that we can subset an <code>ArchRProject</code> to obtain only a select set of cells.</p>
<p>We can subset the project numerically, for example taking the first 100 cells in the project:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="manipulating-an-archrproject.html#cb29-1"></a>projHeme1[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>, ]</span></code></pre></div>
<blockquote>
<p>## class: ArchRProject<br />
## outputDirectory: /oak/stanford/groups/howchang/users/jgranja/ArchRTutorial/ArchRBook/BookOutput4/HemeTutorial<br />
## samples(3): scATAC_BMMC_R1 scATAC_CD34_BMMC_R1 scATAC_PBMC_R1<br />
## sampleColData names(1): ArrowFiles<br />
## cellColData names(11): Sample TSSEnrichment … DoubletScore<br />
## DoubletEnrichment<br />
## numberOfCells(1): 100<br />
## medianTSS(1): 10.7725<br />
## medianFrags(1): 10200.5</p>
</blockquote>
<p>We can subset the project based on certain cell names:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="manipulating-an-archrproject.html#cb30-1"></a>projHeme1[projHeme1<span class="op">$</span>cellNames[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>], ]</span></code></pre></div>
<blockquote>
<p>## class: ArchRProject<br />
## outputDirectory: /oak/stanford/groups/howchang/users/jgranja/ArchRTutorial/ArchRBook/BookOutput4/HemeTutorial<br />
## samples(3): scATAC_BMMC_R1 scATAC_CD34_BMMC_R1 scATAC_PBMC_R1<br />
## sampleColData names(1): ArrowFiles<br />
## cellColData names(11): Sample TSSEnrichment … DoubletScore<br />
## DoubletEnrichment<br />
## numberOfCells(1): 100<br />
## medianTSS(1): 10.7725<br />
## medianFrags(1): 10200.5</p>
</blockquote>
<p>We can subset the project to keep all cells corresponding to a specific sample:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="manipulating-an-archrproject.html#cb31-1"></a>idxSample &lt;-<span class="st"> </span>BiocGenerics<span class="op">::</span><span class="kw">which</span>(projHeme1<span class="op">$</span>Sample <span class="op">%in%</span><span class="st"> &quot;scATAC_BMMC_R1&quot;</span>)</span>
<span id="cb31-2"><a href="manipulating-an-archrproject.html#cb31-2"></a>cellsSample &lt;-<span class="st"> </span>projHeme1<span class="op">$</span>cellNames[idxSample]</span>
<span id="cb31-3"><a href="manipulating-an-archrproject.html#cb31-3"></a>projHeme1[cellsSample, ]</span></code></pre></div>
<blockquote>
<p>## class: ArchRProject<br />
## outputDirectory: /oak/stanford/groups/howchang/users/jgranja/ArchRTutorial/ArchRBook/BookOutput4/HemeTutorial<br />
## samples(3): scATAC_BMMC_R1 scATAC_CD34_BMMC_R1 scATAC_PBMC_R1<br />
## sampleColData names(1): ArrowFiles<br />
## cellColData names(11): Sample TSSEnrichment … DoubletScore<br />
## DoubletEnrichment<br />
## numberOfCells(1): 4932<br />
## medianTSS(1): 15.254<br />
## medianFrags(1): 2771</p>
</blockquote>
<p>We can subset the project to only keep cells that meet a specific cutoff for the TSS enrichment score:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="manipulating-an-archrproject.html#cb32-1"></a>idxPass &lt;-<span class="st"> </span><span class="kw">which</span>(projHeme1<span class="op">$</span>TSSEnrichment <span class="op">&gt;=</span><span class="st"> </span><span class="dv">8</span>)</span>
<span id="cb32-2"><a href="manipulating-an-archrproject.html#cb32-2"></a>cellsPass &lt;-<span class="st"> </span>projHeme1<span class="op">$</span>cellNames[idxPass]</span>
<span id="cb32-3"><a href="manipulating-an-archrproject.html#cb32-3"></a>projHeme1[cellsPass, ]</span></code></pre></div>
<blockquote>
<p>## class: ArchRProject
## outputDirectory: /oak/stanford/groups/howchang/users/jgranja/ArchRTutorial/ArchRBook/BookOutput4/HemeTutorial<br />
## samples(3): scATAC_BMMC_R1 scATAC_CD34_BMMC_R1 scATAC_PBMC_R1<br />
## sampleColData names(1): ArrowFiles<br />
## cellColData names(11): Sample TSSEnrichment … DoubletScore<br />
## DoubletEnrichment<br />
## numberOfCells(1): 10500<br />
## medianTSS(1): 16.9275<br />
## medianFrags(1): 3042</p>
</blockquote>
<p><span style="font-size:20px;font-weight:bold">Example 3. Adding data to an <code>ArchRProject</code></span></p>
<p>We can add columns to <code>cellColData</code> to store any type of cell-specific metadata relevant to our project.</p>
<p>For example, we can add a column to <code>cellColData</code> that contains more legible sample names by removing excess info from the original sample names:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="manipulating-an-archrproject.html#cb33-1"></a>bioNames &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_R2|_R1|scATAC_&quot;</span>,<span class="st">&quot;&quot;</span>,projHeme1<span class="op">$</span>Sample)</span>
<span id="cb33-2"><a href="manipulating-an-archrproject.html#cb33-2"></a><span class="kw">head</span>(bioNames)</span></code></pre></div>
<blockquote>
<p>## [1] “BMMC” “BMMC” “BMMC” “BMMC” “BMMC” “BMMC”<br />
## [7] “BMMC” “BMMC” “BMMC” “BMMC” “BMMC” “BMMC”</p>
</blockquote>
<p>One way to add a column called to <code>cellColData</code> is by directly using the <code>$</code> accessor.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="manipulating-an-archrproject.html#cb34-1"></a>projHeme1<span class="op">$</span>bioNames &lt;-<span class="st"> </span>bioNames</span></code></pre></div>
<p>Alternatively, we can add a column to <code>cellColData</code> using the <code>addCellColData()</code> function. ArchR allows for the addition of columns that only contain information for a subset of cells.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="manipulating-an-archrproject.html#cb35-1"></a>bioNames &lt;-<span class="st"> </span>bioNames[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</span>
<span id="cb35-2"><a href="manipulating-an-archrproject.html#cb35-2"></a>cellNames &lt;-<span class="st"> </span>projHeme1<span class="op">$</span>cellNames[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</span>
<span id="cb35-3"><a href="manipulating-an-archrproject.html#cb35-3"></a>projHeme1 &lt;-<span class="st"> </span><span class="kw">addCellColData</span>(<span class="dt">ArchRProj =</span> projHeme1, <span class="dt">data =</span> <span class="kw">paste0</span>(bioNames),</span>
<span id="cb35-4"><a href="manipulating-an-archrproject.html#cb35-4"></a>    <span class="dt">cells =</span> cellNames, <span class="dt">name =</span> <span class="st">&quot;bioNames2&quot;</span>)</span></code></pre></div>
<p>By default, ArchR will fill in missing entries with <code>NA</code>. Because of this, when we can compare these two columns, we see <code>NA</code> filled in where data wasnt available for <code>bioNames2</code>:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="manipulating-an-archrproject.html#cb36-1"></a><span class="kw">getCellColData</span>(projHeme1, <span class="dt">select =</span> <span class="kw">c</span>(<span class="st">&quot;bioNames&quot;</span>, <span class="st">&quot;bioNames2&quot;</span>))</span></code></pre></div>
<blockquote>
<p>## DataFrame with 10661 rows and 2 columns<br />
## bioNames bioNames2<br />
## <character> <character><br />
## scATAC_BMMC_R1#TTATGTCAGTGATTAG-1 BMMC BMMC<br />
## scATAC_BMMC_R1#AAGATAGTCACCGCGA-1 BMMC BMMC<br />
## scATAC_BMMC_R1#GCATTGAAGATTCCGT-1 BMMC BMMC<br />
## scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1 BMMC BMMC<br />
## scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1 BMMC BMMC<br />
## … … …<br />
## scATAC_PBMC_R1#GCTGCGAAGATCCGAG-1 PBMC NA<br />
## scATAC_PBMC_R1#GCAGCTGGTGGCCTTG-1 PBMC NA<br />
## scATAC_PBMC_R1#GCAGATTGTACGCAAG-1 PBMC NA<br />
## scATAC_PBMC_R1#TTCGTTACATTGAACC-1 PBMC NA<br />
## scATAC_PBMC_R1#CGCTATCGTGAGGTCA-1 PBMC NA</p>
</blockquote>
<p><span style="font-size:20px;font-weight:bold">Example 4. Obtaining columns from <code>cellColData</code></span></p>
<p>ArchR provides the <code>getCellColData()</code> function to enable easy retreival of metadata columns from an <code>ArchRProject</code>.</p>
<p>For example, we can retrieve a column by name, such as the number of unique nuclear (i.e. non-mitochondrial) fragments per cell:</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="manipulating-an-archrproject.html#cb37-1"></a>df &lt;-<span class="st"> </span><span class="kw">getCellColData</span>(projHeme1, <span class="dt">select =</span> <span class="st">&quot;nFrags&quot;</span>)</span>
<span id="cb37-2"><a href="manipulating-an-archrproject.html#cb37-2"></a>df</span></code></pre></div>
<blockquote>
<p>## DataFrame with 10661 rows and 1 column<br />
## nFrags<br />
## <numeric><br />
## scATAC_BMMC_R1#TTATGTCAGTGATTAG-1 26189<br />
## scATAC_BMMC_R1#AAGATAGTCACCGCGA-1 20648<br />
## scATAC_BMMC_R1#GCATTGAAGATTCCGT-1 18991<br />
## scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1 18296<br />
## scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1 17458<br />
## … …<br />
## scATAC_PBMC_R1#GCTGCGAAGATCCGAG-1 1038<br />
## scATAC_PBMC_R1#GCAGCTGGTGGCCTTG-1 1037<br />
## scATAC_PBMC_R1#GCAGATTGTACGCAAG-1 1033<br />
## scATAC_PBMC_R1#TTCGTTACATTGAACC-1 1033<br />
## scATAC_PBMC_R1#CGCTATCGTGAGGTCA-1 1002</p>
</blockquote>
<p>Instead of selecting a column by name, we can actually perform operations on a given column using its column name:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="manipulating-an-archrproject.html#cb38-1"></a>df &lt;-<span class="st"> </span><span class="kw">getCellColData</span>(projHeme1, <span class="dt">select =</span> <span class="kw">c</span>(<span class="st">&quot;log10(nFrags)&quot;</span>, <span class="st">&quot;nFrags - 1&quot;</span>))</span>
<span id="cb38-2"><a href="manipulating-an-archrproject.html#cb38-2"></a>df</span></code></pre></div>
<blockquote>
<p>## DataFrame with 10661 rows and 2 columns<br />
## log10(nFrags) nFrags - 1<br />
## <numeric> <numeric><br />
## scATAC_BMMC_R1#TTATGTCAGTGATTAG-1 4.4181189156542 26188<br />
## scATAC_BMMC_R1#AAGATAGTCACCGCGA-1 4.31487799153581 20647<br />
## scATAC_BMMC_R1#GCATTGAAGATTCCGT-1 4.27854783377585 18990<br />
## scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1 4.26235615159869 18295<br />
## scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1 4.24199448915678 17457<br />
## … … …<br />
## scATAC_PBMC_R1#GCTGCGAAGATCCGAG-1 3.01619735351244 1037<br />
## scATAC_PBMC_R1#GCAGCTGGTGGCCTTG-1 3.01577875638904 1036<br />
## scATAC_PBMC_R1#GCAGATTGTACGCAAG-1 3.01410032151962 1032<br />
## scATAC_PBMC_R1#TTCGTTACATTGAACC-1 3.01410032151962 1032<br />
## scATAC_PBMC_R1#CGCTATCGTGAGGTCA-1 3.00086772153123 1001</p>
</blockquote>
<p><span style="font-size:20px;font-weight:bold">Example 5. Plotting QC metrics - log10(Unique Fragments) vs TSS enrichment score</span></p>
<p>Repeating the example shown above, we can easily obtain standard scATAC-seq metrics for quality control of individual cells. We have found that the most robust metrics for quality control are the TSS enrichment score (a measure of signal-to-background in ATAC-seq data) and the number of unique nuclear fragments (because cells with very few fragments do not have enough data to confidently analyze).</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="manipulating-an-archrproject.html#cb39-1"></a>df &lt;-<span class="st"> </span><span class="kw">getCellColData</span>(projHeme1, <span class="dt">select =</span> <span class="kw">c</span>(<span class="st">&quot;log10(nFrags)&quot;</span>, <span class="st">&quot;TSSEnrichment&quot;</span>))</span>
<span id="cb39-2"><a href="manipulating-an-archrproject.html#cb39-2"></a>df</span></code></pre></div>
<blockquote>
<p>## DataFrame with 10661 rows and 2 columns<br />
## log10(nFrags) TSSEnrichment<br />
## <numeric> <numeric><br />
## scATAC_BMMC_R1#TTATGTCAGTGATTAG-1 4.4181189156542 7.149<br />
## scATAC_BMMC_R1#AAGATAGTCACCGCGA-1 4.31487799153581 7.911<br />
## scATAC_BMMC_R1#GCATTGAAGATTCCGT-1 4.27854783377585 4.505<br />
## scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1 4.26235615159869 6.946<br />
## scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1 4.24199448915678 4.799<br />
## … … …<br />
## scATAC_PBMC_R1#GCTGCGAAGATCCGAG-1 3.01619735351244 24.356<br />
## scATAC_PBMC_R1#GCAGCTGGTGGCCTTG-1 3.01577875638904 22.537<br />
## scATAC_PBMC_R1#GCAGATTGTACGCAAG-1 3.01410032151962 20.146<br />
## scATAC_PBMC_R1#TTCGTTACATTGAACC-1 3.01410032151962 30.198<br />
## scATAC_PBMC_R1#CGCTATCGTGAGGTCA-1 3.00086772153123 21.485</p>
</blockquote>
<p>Now lets plot the number of unique nuclear fragments (log10) by the TSS enrichment score. This type of plot is
key for identifying high quality cells. You’ll notice that the cutoffs that we previously specified when creating the Arrow files (via <code>filterTSS</code> and <code>filterFrags</code>) have already removed low quality cells. However, if we noticed that the previously applied QC filters were not adequate for this sample, we could further adjust our cutoffs based on this plot or re-generate the Arrow files if needed.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="manipulating-an-archrproject.html#cb40-1"></a>p &lt;-<span class="st"> </span><span class="kw">ggPoint</span>(</span>
<span id="cb40-2"><a href="manipulating-an-archrproject.html#cb40-2"></a>    <span class="dt">x =</span> df[,<span class="dv">1</span>], </span>
<span id="cb40-3"><a href="manipulating-an-archrproject.html#cb40-3"></a>    <span class="dt">y =</span> df[,<span class="dv">2</span>], </span>
<span id="cb40-4"><a href="manipulating-an-archrproject.html#cb40-4"></a>    <span class="dt">colorDensity =</span> <span class="ot">TRUE</span>,</span>
<span id="cb40-5"><a href="manipulating-an-archrproject.html#cb40-5"></a>    <span class="dt">continuousSet =</span> <span class="st">&quot;sambaNight&quot;</span>,</span>
<span id="cb40-6"><a href="manipulating-an-archrproject.html#cb40-6"></a>    <span class="dt">xlabel =</span> <span class="st">&quot;Log10 Unique Fragments&quot;</span>,</span>
<span id="cb40-7"><a href="manipulating-an-archrproject.html#cb40-7"></a>    <span class="dt">ylabel =</span> <span class="st">&quot;TSS Enrichment&quot;</span>,</span>
<span id="cb40-8"><a href="manipulating-an-archrproject.html#cb40-8"></a>    <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="kw">log10</span>(<span class="dv">500</span>), <span class="kw">quantile</span>(df[,<span class="dv">1</span>], <span class="dt">probs =</span> <span class="fl">0.99</span>)),</span>
<span id="cb40-9"><a href="manipulating-an-archrproject.html#cb40-9"></a>    <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">quantile</span>(df[,<span class="dv">2</span>], <span class="dt">probs =</span> <span class="fl">0.99</span>))</span>
<span id="cb40-10"><a href="manipulating-an-archrproject.html#cb40-10"></a>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">4</span>, <span class="dt">lty =</span> <span class="st">&quot;dashed&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> <span class="dv">3</span>, <span class="dt">lty =</span> <span class="st">&quot;dashed&quot;</span>)</span>
<span id="cb40-11"><a href="manipulating-an-archrproject.html#cb40-11"></a></span>
<span id="cb40-12"><a href="manipulating-an-archrproject.html#cb40-12"></a>p</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/TSS-vs-Frags_1.png" width="500" height="500" /></p>
<p>To save an editable vectorized version of this plot, we use <code>plotPDF()</code>.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="manipulating-an-archrproject.html#cb41-1"></a><span class="kw">plotPDF</span>(p, <span class="dt">name =</span> <span class="st">&quot;TSS-vs-Frags.pdf&quot;</span>, <span class="dt">ArchRProj =</span> projHeme1, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## [1] 0</p>
</blockquote>
<p><a href="images/HemeWalkthrough/PDF/TSS-vs-Frags.pdf">Download PDF : TSS-vs-Frags.pdf</a></p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="creating-an-archrproject-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="plotting-sample-statistics-from-an-archrproject.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/03_ArchRProject.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/03_ArchRProject.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
},
"search": true,
"info": false
});
});
</script>

</body>

</html>
