<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>12.1 Motif Enrichment in Differential Peaks | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="12.1 Motif Enrichment in Differential Peaks | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="12.1 Motif Enrichment in Differential Peaks | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-05-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="motif-and-feature-enrichment-with-archr.html"/>
<link rel="next" href="motif-enrichment-in-marker-peaks.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://www.archrproject.com/index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li><a href="index.html#section"><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>1.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="1.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>1.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="1.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>1.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>1.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="1.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="1.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>1.7</b> Per-cell Quality Control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>2.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="2.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="3.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>4.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="4.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>4.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="4.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>5.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>6.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="6.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>6.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="6.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>6.3</b> Dimensionality Reduction After Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>7</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>7.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="7.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>7.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="7.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>7.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="7.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>7.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="7.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="7.6" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.6</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.7" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.7</b> Launching the ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>8</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>8.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>9</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="9.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>10.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="10.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>10.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="10.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>10.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="10.4" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>10.4</b> Add Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>11</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>11.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="11.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>11.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="11.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>11.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-enrichment-in-differential-peaks.html"><a href="motif-enrichment-in-differential-peaks.html"><i class="fa fa-check"></i><b>12.1</b> Motif Enrichment in Differential Peaks</a></li>
<li class="chapter" data-level="12.2" data-path="motif-enrichment-in-marker-peaks.html"><a href="motif-enrichment-in-marker-peaks.html"><i class="fa fa-check"></i><b>12.2</b> Motif Enrichment in Marker Peaks</a></li>
<li class="chapter" data-level="12.3" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>12.3</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="12.4" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>12.4</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>13</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>13.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="13.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>13.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>14</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>14.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="14.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>14.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="14.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>14.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>15</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>15.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="15.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>15.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="15.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>15.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="15.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>15.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>16</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="myeloid-trajectory-monocyte-differentiation.html"><a href="myeloid-trajectory-monocyte-differentiation.html"><i class="fa fa-check"></i><b>16.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="16.2" data-path="lymphoid-trajectory-b-cell-cifferentiation.html"><a href="lymphoid-trajectory-b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>16.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>17</b> Plot aesthetics in ArchR</a></li>
<li class="chapter" data-level="18" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>18</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="index.html">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="motif-enrichment-in-differential-peaks" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Motif Enrichment in Differential Peaks</h2>
<p>Continuing our analysis of differential peaks from the previous chapter, we can look for motifs that are enriched in peaks that are up or down in various cell types. To do this, we must first add these motif annotations to our <code>ArchRProject</code>. This effectively creates a binary matrix where the presence of a motif in each peak is indicated numerically. We do this using the <code>addMotifAnnotations()</code> function which determines motif presence in the peak set stored in the <code>ArchRProject</code>.</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="motif-enrichment-in-differential-peaks.html#cb190-1"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addMotifAnnotations</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">motifSet =</span> <span class="st">&quot;cisbp&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;Motif&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## No methods found in package ‘IRanges’ for request: ‘score’ when loading ‘TFBSTools’<br />
## 2020-04-15 10:38:11 : Gettting Motif Set, Species : Homo sapiens, 0 mins elapsed.<br />
## Using version 2 motifs!<br />
## 2020-04-15 10:38:13 : Finding Motif Positions with motifmatchr!, 0.027 mins elapsed.<br />
## 2020-04-15 10:41:40 : Creating Motif Overlap Matrix, 3.48 mins elapsed.<br />
## 2020-04-15 10:41:43 : Finished Getting Motif Info!, 3.523 mins elapsed.</p>
</blockquote>
<p>We can then use the differential testing <code>SummarizedExperiment</code> object <code>markerTest</code> which was generated in the previous chapter to define the set of significantly differential peaks that we are interested in testing for motif enrichment. In this case, we are looking for peaks that have an <code>FDR &lt;= 0.1</code> and a <code>Log2FC &gt;= 0.5</code>. In the context of the differential comparison made in <code>markerTest</code>, these represnt peaks that are more accessible in “Erythroid” cells than “Progenitor” cells. We can test these differentially accessible peaks for enrichment of various motifs using the <code>peakAnnoEnrichment()</code> function. This function is a generalizable function that can be used for many different enrichment tests as we will demonstrate throughout this chapter.</p>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="motif-enrichment-in-differential-peaks.html#cb191-1"></a>motifsUp &lt;-<span class="st"> </span><span class="kw">peakAnnoEnrichment</span>(</span>
<span id="cb191-2"><a href="motif-enrichment-in-differential-peaks.html#cb191-2"></a>    <span class="dt">seMarker =</span> markerTest,</span>
<span id="cb191-3"><a href="motif-enrichment-in-differential-peaks.html#cb191-3"></a>    <span class="dt">ArchRProj =</span> projHeme5,</span>
<span id="cb191-4"><a href="motif-enrichment-in-differential-peaks.html#cb191-4"></a>    <span class="dt">peakAnnotation =</span> <span class="st">&quot;Motif&quot;</span>,</span>
<span id="cb191-5"><a href="motif-enrichment-in-differential-peaks.html#cb191-5"></a>    <span class="dt">cutOff =</span> <span class="st">&quot;FDR &lt;= 0.1 &amp; Log2FC &gt;= 0.5&quot;</span></span>
<span id="cb191-6"><a href="motif-enrichment-in-differential-peaks.html#cb191-6"></a>  )</span></code></pre></div>
<blockquote>
<p>## 2020-04-15 10:41:49 : Computing Enrichments 1 of 1, 0.066 mins elapsed.</p>
</blockquote>
<p>The output of <code>peakAnnoEnrichment()</code> is a <code>SummarizedExperiment</code> object containing multiple <code>assays</code> that store the results of enrichment testing with the hypergeometric test.</p>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="motif-enrichment-in-differential-peaks.html#cb192-1"></a>motifsUp</span></code></pre></div>
<blockquote>
<p>## class: SummarizedExperiment<br />
## dim: 870 1<br />
## metadata(0):<br />
## assays(10): mlog10Padj mlog10p … CompareFrequency feature<br />
## rownames(870): TFAP2B_1 TFAP2D_2 … TBX18_869 TBX22_870<br />
## rowData names(0):<br />
## colnames(1): Erythroid<br />
## colData names(0):</p>
</blockquote>
<p>To prepare this data for plotting with <code>ggplot</code> we can create a simplified <code>data.frame</code> object containing the motif names, the corrected p-values, and the significance rank.</p>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="motif-enrichment-in-differential-peaks.html#cb193-1"></a>df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">TF =</span> <span class="kw">rownames</span>(motifsUp), <span class="dt">mlog10Padj =</span> <span class="kw">assay</span>(motifsUp)[,<span class="dv">1</span>])</span>
<span id="cb193-2"><a href="motif-enrichment-in-differential-peaks.html#cb193-2"></a>df &lt;-<span class="st"> </span>df[<span class="kw">order</span>(df<span class="op">$</span>mlog10Padj, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>),]</span>
<span id="cb193-3"><a href="motif-enrichment-in-differential-peaks.html#cb193-3"></a>df<span class="op">$</span>rank &lt;-<span class="st"> </span><span class="kw">seq_len</span>(<span class="kw">nrow</span>(df))</span></code></pre></div>
<p>As expected, the most enriched motifs in the peaks that are more accessible in “Erythroid” cells correspond to GATA transcription factors, consistent with the well-studied role of GATA1 in erythroid differentiation.</p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="motif-enrichment-in-differential-peaks.html#cb194-1"></a><span class="kw">head</span>(df)</span></code></pre></div>
<blockquote>
<p>## TF mlog10Padj rank<br />
## 388 GATA2_388 535.7098 1<br />
## 383 GATA1_383 523.8809 2<br />
## 384 GATA3_384 421.6064 3<br />
## 385 GATA5_385 407.2538 4<br />
## 386 GATA4_386 309.7756 5<br />
## 387 GATA6_387 216.7223 6</p>
</blockquote>
<p>Using <code>ggplot</code> we can plot the rank-sorted TF motifs and color them by the significance of their enrichment. Here we use <code>ggrepel</code> to label each TF motif.</p>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="motif-enrichment-in-differential-peaks.html#cb195-1"></a>ggUp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(rank, mlog10Padj, <span class="dt">color =</span> mlog10Padj)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb195-2"><a href="motif-enrichment-in-differential-peaks.html#cb195-2"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb195-3"><a href="motif-enrichment-in-differential-peaks.html#cb195-3"></a><span class="st">  </span>ggrepel<span class="op">::</span><span class="kw">geom_label_repel</span>(</span>
<span id="cb195-4"><a href="motif-enrichment-in-differential-peaks.html#cb195-4"></a>        <span class="dt">data =</span> df[<span class="kw">rev</span>(<span class="kw">seq_len</span>(<span class="dv">30</span>)), ], <span class="kw">aes</span>(<span class="dt">x =</span> rank, <span class="dt">y =</span> mlog10Padj, <span class="dt">label =</span> TF), </span>
<span id="cb195-5"><a href="motif-enrichment-in-differential-peaks.html#cb195-5"></a>        <span class="dt">size =</span> <span class="fl">1.5</span>,</span>
<span id="cb195-6"><a href="motif-enrichment-in-differential-peaks.html#cb195-6"></a>        <span class="dt">nudge_x =</span> <span class="dv">2</span>,</span>
<span id="cb195-7"><a href="motif-enrichment-in-differential-peaks.html#cb195-7"></a>        <span class="dt">color =</span> <span class="st">&quot;black&quot;</span></span>
<span id="cb195-8"><a href="motif-enrichment-in-differential-peaks.html#cb195-8"></a>  ) <span class="op">+</span><span class="st"> </span><span class="kw">theme_ArchR</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb195-9"><a href="motif-enrichment-in-differential-peaks.html#cb195-9"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;-log10(P-adj) Motif Enrichment&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb195-10"><a href="motif-enrichment-in-differential-peaks.html#cb195-10"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Rank Sorted TFs Enriched&quot;</span>) <span class="op">+</span></span>
<span id="cb195-11"><a href="motif-enrichment-in-differential-peaks.html#cb195-11"></a><span class="st">  </span><span class="kw">scale_color_gradientn</span>(<span class="dt">colors =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;comet&quot;</span>))</span>
<span id="cb195-12"><a href="motif-enrichment-in-differential-peaks.html#cb195-12"></a></span>
<span id="cb195-13"><a href="motif-enrichment-in-differential-peaks.html#cb195-13"></a>ggUp</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Erythroid-vs-Progenitor-Markers-Motifs-Enriched_1.png" width="600" height="600" /></p>
<p>We can perform the same analyses for the peaks that are more accessible in the “Progenitor” cells by using peaks with <code>Log2FC &lt;= -0.5</code>.</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="motif-enrichment-in-differential-peaks.html#cb196-1"></a>motifsDo &lt;-<span class="st"> </span><span class="kw">peakAnnoEnrichment</span>(</span>
<span id="cb196-2"><a href="motif-enrichment-in-differential-peaks.html#cb196-2"></a>    <span class="dt">seMarker =</span> markerTest,</span>
<span id="cb196-3"><a href="motif-enrichment-in-differential-peaks.html#cb196-3"></a>    <span class="dt">ArchRProj =</span> projHeme5,</span>
<span id="cb196-4"><a href="motif-enrichment-in-differential-peaks.html#cb196-4"></a>    <span class="dt">peakAnnotation =</span> <span class="st">&quot;Motif&quot;</span>,</span>
<span id="cb196-5"><a href="motif-enrichment-in-differential-peaks.html#cb196-5"></a>    <span class="dt">cutOff =</span> <span class="st">&quot;FDR &lt;= 0.1 &amp; Log2FC &lt;= -0.5&quot;</span></span>
<span id="cb196-6"><a href="motif-enrichment-in-differential-peaks.html#cb196-6"></a>  )</span></code></pre></div>
<blockquote>
<p>## 2020-04-15 10:42:01 : Computing Enrichments 1 of 1, 0.053 mins elapsed.</p>
</blockquote>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="motif-enrichment-in-differential-peaks.html#cb197-1"></a>motifsDo</span></code></pre></div>
<blockquote>
<p>## class: SummarizedExperiment<br />
## dim: 870 1<br />
## metadata(0):<br />
## assays(10): mlog10Padj mlog10p … CompareFrequency feature<br />
## rownames(870): TFAP2B_1 TFAP2D_2 … TBX18_869 TBX22_870<br />
## rowData names(0):<br />
## colnames(1): Erythroid<br />
## colData names(0):</p>
</blockquote>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="motif-enrichment-in-differential-peaks.html#cb198-1"></a>df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">TF =</span> <span class="kw">rownames</span>(motifsDo), <span class="dt">mlog10Padj =</span> <span class="kw">assay</span>(motifsDo)[,<span class="dv">1</span>])</span>
<span id="cb198-2"><a href="motif-enrichment-in-differential-peaks.html#cb198-2"></a>df &lt;-<span class="st"> </span>df[<span class="kw">order</span>(df<span class="op">$</span>mlog10Padj, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>),]</span>
<span id="cb198-3"><a href="motif-enrichment-in-differential-peaks.html#cb198-3"></a>df<span class="op">$</span>rank &lt;-<span class="st"> </span><span class="kw">seq_len</span>(<span class="kw">nrow</span>(df))</span></code></pre></div>
<p>In this case, the most enriched motifs in the peaks that are more accessible in “Progenitor” cells correspond to RUNX, ELF, and CBFB motifs.</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="motif-enrichment-in-differential-peaks.html#cb199-1"></a><span class="kw">head</span>(df)</span></code></pre></div>
<blockquote>
<p>## TF mlog10Padj rank<br />
## 326 ELF2_326 88.68056 1<br />
## 733 RUNX1_733 64.00586 2<br />
## 801 CBFB_801 53.55426 3<br />
## 732 RUNX2_732 53.14766 4<br />
## 734 ENSG00000250096_734 53.14766 5<br />
## 336 SPIB_336 52.79666 6</p>
</blockquote>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="motif-enrichment-in-differential-peaks.html#cb200-1"></a>ggDo &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(rank, mlog10Padj, <span class="dt">color =</span> mlog10Padj)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb200-2"><a href="motif-enrichment-in-differential-peaks.html#cb200-2"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb200-3"><a href="motif-enrichment-in-differential-peaks.html#cb200-3"></a><span class="st">  </span>ggrepel<span class="op">::</span><span class="kw">geom_label_repel</span>(</span>
<span id="cb200-4"><a href="motif-enrichment-in-differential-peaks.html#cb200-4"></a>        <span class="dt">data =</span> df[<span class="kw">rev</span>(<span class="kw">seq_len</span>(<span class="dv">30</span>)), ], <span class="kw">aes</span>(<span class="dt">x =</span> rank, <span class="dt">y =</span> mlog10Padj, <span class="dt">label =</span> TF), </span>
<span id="cb200-5"><a href="motif-enrichment-in-differential-peaks.html#cb200-5"></a>        <span class="dt">size =</span> <span class="fl">1.5</span>,</span>
<span id="cb200-6"><a href="motif-enrichment-in-differential-peaks.html#cb200-6"></a>        <span class="dt">nudge_x =</span> <span class="dv">2</span>,</span>
<span id="cb200-7"><a href="motif-enrichment-in-differential-peaks.html#cb200-7"></a>        <span class="dt">color =</span> <span class="st">&quot;black&quot;</span></span>
<span id="cb200-8"><a href="motif-enrichment-in-differential-peaks.html#cb200-8"></a>  ) <span class="op">+</span><span class="st"> </span><span class="kw">theme_ArchR</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb200-9"><a href="motif-enrichment-in-differential-peaks.html#cb200-9"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;-log10(FDR) Motif Enrichment&quot;</span>) <span class="op">+</span></span>
<span id="cb200-10"><a href="motif-enrichment-in-differential-peaks.html#cb200-10"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Rank Sorted TFs Enriched&quot;</span>) <span class="op">+</span></span>
<span id="cb200-11"><a href="motif-enrichment-in-differential-peaks.html#cb200-11"></a><span class="st">  </span><span class="kw">scale_color_gradientn</span>(<span class="dt">colors =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;comet&quot;</span>))</span>
<span id="cb200-12"><a href="motif-enrichment-in-differential-peaks.html#cb200-12"></a></span>
<span id="cb200-13"><a href="motif-enrichment-in-differential-peaks.html#cb200-13"></a>ggDo</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Erythroid-vs-Progenitor-Markers-Motifs-Enriched_2.png" width="600" height="600" /></p>
<p>To save an editable vectorized version of these plots, we use the <code>plotPDF()</code> function.</p>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="motif-enrichment-in-differential-peaks.html#cb201-1"></a><span class="kw">plotPDF</span>(ggUp, ggDo, <span class="dt">name =</span> <span class="st">&quot;Erythroid-vs-Progenitor-Markers-Motifs-Enriched&quot;</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## [1] “plotting ggplot!”<br />
## [1] 0</p>
</blockquote>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="motif-and-feature-enrichment-with-archr.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="motif-enrichment-in-marker-peaks.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/12_PeakEnrichment.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/12_PeakEnrichment.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
},
"search": true,
"info": false
});
});
</script>

</body>

</html>
