<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>16.1 Myeloid Trajectory - Monocyte Differentiation | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="16.1 Myeloid Trajectory - Monocyte Differentiation | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="16.1 Myeloid Trajectory - Monocyte Differentiation | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-05-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="trajectory-analysis-with-archr.html"/>
<link rel="next" href="lymphoid-trajectory-b-cell-cifferentiation.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://www.archrproject.com/index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li><a href="index.html#section"><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>1.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="1.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>1.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="1.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>1.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>1.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="1.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="1.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>1.7</b> Per-cell Quality Control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>2.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="2.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="3.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>4.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="4.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>4.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="4.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>5.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>6.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="6.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>6.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="6.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>6.3</b> Dimensionality Reduction After Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>7</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>7.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="7.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>7.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="7.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>7.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="7.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>7.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="7.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="7.6" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.6</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.7" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.7</b> Launching the ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>8</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>8.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>9</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="9.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>10.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="10.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>10.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="10.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>10.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="10.4" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>10.4</b> Add Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>11</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>11.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="11.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>11.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="11.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>11.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-enrichment-in-differential-peaks.html"><a href="motif-enrichment-in-differential-peaks.html"><i class="fa fa-check"></i><b>12.1</b> Motif Enrichment in Differential Peaks</a></li>
<li class="chapter" data-level="12.2" data-path="motif-enrichment-in-marker-peaks.html"><a href="motif-enrichment-in-marker-peaks.html"><i class="fa fa-check"></i><b>12.2</b> Motif Enrichment in Marker Peaks</a></li>
<li class="chapter" data-level="12.3" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>12.3</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="12.4" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>12.4</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>13</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>13.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="13.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>13.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>14</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>14.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="14.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>14.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="14.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>14.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>15</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>15.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="15.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>15.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="15.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>15.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="15.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>15.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>16</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="myeloid-trajectory-monocyte-differentiation.html"><a href="myeloid-trajectory-monocyte-differentiation.html"><i class="fa fa-check"></i><b>16.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="16.2" data-path="lymphoid-trajectory-b-cell-cifferentiation.html"><a href="lymphoid-trajectory-b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>16.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>17</b> Plot aesthetics in ArchR</a></li>
<li class="chapter" data-level="18" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>18</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="index.html">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="myeloid-trajectory---monocyte-differentiation" class="section level2" number="16.1">
<h2><span class="header-section-number">16.1</span> Myeloid Trajectory - Monocyte Differentiation</h2>
<p>In this section, we will create a cellular trajectory that approximates the differentiation of HSCs into fully differentiated monocytes. To start, lets review the clusters and cell types that we defined previously, stored in <code>cellColData</code> in columns named “Clusters” and “Clusters2”. Overlaying these cell groupings on our UMAP embedding shows the different cell types that we are interested in.</p>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb324-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;Clusters&quot;</span>, <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-15e1326ab29a8-Date-2020-04-15_Time-14-59-47.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = cellColData<br />
## Plotting Embedding<br />
## 1<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-15e1326ab29a8-Date-2020-04-15_Time-14-59-47.log</p>
</blockquote>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb325-1"></a>p2 &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;Clusters2&quot;</span>, <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-15e1339bfbfcf-Date-2020-04-15_Time-14-59-49.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = cellColData<br />
## Plotting Embedding<br />
## 1<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-15e1339bfbfcf-Date-2020-04-15_Time-14-59-49.log</p>
</blockquote>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb326-1"></a><span class="kw">ggAlignPlots</span>(p1, p2, <span class="dt">type =</span> <span class="st">&quot;h&quot;</span>)</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-Clusters12-Combined_1.png" width="800" height="400" /></p>
<div id="pseudo-time-umaps-and-individual-feature-plots" class="section level3" number="16.1.1">
<h3><span class="header-section-number">16.1.1</span> Pseudo-time UMAPs and individual feature plots</h3>
<p>We will use the cell type definitions that are stored in “Clusters2”. As mentioned above, we are creating a trajectory from stem cells (“Progenitor”), through the commited myeloid progenitor (“GMP”), to monocytes (“Mono”). The first step in creating a trajectory is to create a trajectory backbone in the form of an ordered vector of cell group labels.</p>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb327-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb327-1"></a>trajectory &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Progenitor&quot;</span>, <span class="st">&quot;GMP&quot;</span>, <span class="st">&quot;Mono&quot;</span>)</span>
<span id="cb327-2"><a href="myeloid-trajectory-monocyte-differentiation.html#cb327-2"></a>trajectory</span></code></pre></div>
<blockquote>
<p>## [1] “Progenitor” “GMP” “Mono”</p>
</blockquote>
<p>We use the <code>addTrajectory()</code> function to create a trajectory and we add this to our <code>ArchRProject</code>. We will call this trajectory “MyeloidU”. What this does is it creates a new column in <code>cellColData</code> called “MyeloidU” that stores the pseudo-time value for each cell in the trajectory. Cells that are not part of the trajectory are labeled with <code>NA</code>.</p>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb328-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb328-1"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addTrajectory</span>(</span>
<span id="cb328-2"><a href="myeloid-trajectory-monocyte-differentiation.html#cb328-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb328-3"><a href="myeloid-trajectory-monocyte-differentiation.html#cb328-3"></a>    <span class="dt">name =</span> <span class="st">&quot;MyeloidU&quot;</span>, </span>
<span id="cb328-4"><a href="myeloid-trajectory-monocyte-differentiation.html#cb328-4"></a>    <span class="dt">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>,</span>
<span id="cb328-5"><a href="myeloid-trajectory-monocyte-differentiation.html#cb328-5"></a>    <span class="dt">trajectory =</span> trajectory, </span>
<span id="cb328-6"><a href="myeloid-trajectory-monocyte-differentiation.html#cb328-6"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>, </span>
<span id="cb328-7"><a href="myeloid-trajectory-monocyte-differentiation.html#cb328-7"></a>    <span class="dt">force =</span> <span class="ot">TRUE</span></span>
<span id="cb328-8"><a href="myeloid-trajectory-monocyte-differentiation.html#cb328-8"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-addTrajectory-15e132f14b12b-Date-2020-04-15_Time-15-00-01.log<br />
## If there is an issue, please report to github with logFile!<br />
## Filtering outliers<br />
## Initial Alignment Before Spline Fit<br />
## Spline Fit<br />
## KNN to Spline<br />
## Aligning cells not in trajectory<br />
## ArchR logging successful to : ArchRLogs/ArchR-addTrajectory-15e132f14b12b-Date-2020-04-15_Time-15-00-01.log</p>
</blockquote>
<p>We can look at this information and see that each cell has a unique pseudotime value between 0 and 100. We exclude cells with <code>NA</code> values because these are not part of the trajectory.</p>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb329-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb329-1"></a><span class="kw">head</span>(projHeme5<span class="op">$</span>MyeloidU[<span class="op">!</span><span class="kw">is.na</span>(projHeme5<span class="op">$</span>MyeloidU)])</span></code></pre></div>
<blockquote>
<p>## [1] 46.07479 53.75027 44.82834 43.18828 47.49617 43.21015</p>
</blockquote>
<p>To plot this trajectory, we use the <code>plotTrajectory()</code> function which overlays the pseudo-time values on our UMAP embedding and displays an arrow approximating the trajectory path from the spline-fit. Cells that are not part of the trajectory are colored gray. In this example, we use <code>colorBy = "cellColData"</code> to tell ArchR to look within <code>cellColData</code> for the column specified by <code>name</code> - in this case, the “MyeloidU” pseudo-time trajectory. While it may seem conterintuitive to list “MyeloidU” for both <code>trajectory</code> and <code>name</code>, this is done because <code>trajectory</code> tells ArchR which subset of cells we are interested in and <code>name</code> tells ArchR how to color that subset of cells.</p>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb330-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb330-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotTrajectory</span>(projHeme5, <span class="dt">trajectory =</span> <span class="st">&quot;MyeloidU&quot;</span>, <span class="dt">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;MyeloidU&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-15e136ad2f619-Date-2020-04-15_Time-15-00-03.log<br />
## If there is an issue, please report to github with logFile!<br />
## Plotting<br />
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).<br />
## Plotting Trajectory<br />
## Adding Inferred Arrow Trajectory to Plot<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-15e136ad2f619-Date-2020-04-15_Time-15-00-03.log</p>
</blockquote>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb331-1"></a>p[[<span class="dv">1</span>]]</span></code></pre></div>
<blockquote>
<p>## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Plot-MyeloidU-Traj-UMAP_1.png" width="500" height="500" /></p>
<p>To save an editable vectorized version of this plot, we use <code>plotPDF()</code>.</p>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb332-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb332-1"></a><span class="kw">plotPDF</span>(p, <span class="dt">name =</span> <span class="st">&quot;Plot-MyeloidU-Traj-UMAP.pdf&quot;</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).<br />
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).<br />
## [1] “plotting ggplot!”<br />
## <code>geom_smooth()</code> using method = ‘gam’ and formula ‘y ~ s(x, bs = “cs”)’<br />
## <code>geom_smooth()</code> using method = ‘gam’ and formula ‘y ~ s(x, bs = “cs”)’<br />
## [1] 0</p>
</blockquote>
<p>We can also overlay other features on the trajectory within our UMAP embedding. This allows us to display specific features only within the cells that are relevant to our trajectory.</p>
<p>If you have not already added impute weights to your <code>projHeme5</code> project, lets do that now.</p>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb333-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb333-1"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addImputeWeights</span>(projHeme5)</span></code></pre></div>
<blockquote>
<p>## 2020-04-15 15:00:28 : Computing Impute Weights Using Magic (Cell 2018), 0 mins elapsed.<br />
## 2020-04-15 15:00:40 : Completed Getting Magic Weights!, 0.206 mins elapsed.</p>
</blockquote>
<p>Then, we can plot the “MyeloidU” trajectory but color the cells by the gene score value of the CEBPB gene, a known regulator of monocyte function that becomes active during the differentiation process. We indicate the matrix to use via the <code>colorBy</code> parameter and the feature to use via the <code>name</code> parameter.</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb334-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotTrajectory</span>(projHeme5, <span class="dt">trajectory =</span> <span class="st">&quot;MyeloidU&quot;</span>, <span class="dt">colorBy =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;CEBPB&quot;</span>, <span class="dt">continuousSet =</span> <span class="st">&quot;horizonExtra&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-15e13179524ae-Date-2020-04-15_Time-15-00-40.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting<br />
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).<br />
## Plotting Trajectory<br />
## Adding Inferred Arrow Trajectory to Plot<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-15e13179524ae-Date-2020-04-15_Time-15-00-40.log</p>
</blockquote>
<p>We can repeat this process but color the cells by their linked gene expression via the <code>GeneIntegrationMatrix</code>.</p>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb335-1"></a>p2 &lt;-<span class="st"> </span><span class="kw">plotTrajectory</span>(projHeme5, <span class="dt">trajectory =</span> <span class="st">&quot;MyeloidU&quot;</span>, <span class="dt">colorBy =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;CEBPB&quot;</span>, <span class="dt">continuousSet =</span> <span class="st">&quot;blueYellow&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-15e133644196a-Date-2020-04-15_Time-15-00-47.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting<br />
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).<br />
## Plotting Trajectory<br />
## Adding Inferred Arrow Trajectory to Plot<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-15e133644196a-Date-2020-04-15_Time-15-00-47.log</p>
</blockquote>
<p>The <code>plotTrajectory()</code> function actually returns a list of relevant plots. The first plot in the list is a UMAP embedding, colorized as specified in the function call.</p>
<p>Comparing these UMAP plots side-by-side for the gene score and gene expression, we see that the activity of the CEBPB gene is highly specific to monocyte cells in the later part of the pseudo-time trajectory.</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb336-1"></a><span class="kw">ggAlignPlots</span>(p1[[<span class="dv">1</span>]], p2[[<span class="dv">1</span>]], <span class="dt">type =</span> <span class="st">&quot;h&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).<br />
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-CEBPB-Combined_1.png" width="800" height="400" /></p>
<p>The second plot that is returned by <code>plotTrajectory()</code> is a dot plot of pseudo-time versus the value of the relevant feature, in this case, the gene score or gene expression of CEBPB. In this case, the cells are colored by their pseudo-time.</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb337-1"></a><span class="kw">ggAlignPlots</span>(p1[[<span class="dv">2</span>]], p2[[<span class="dv">2</span>]], <span class="dt">type =</span> <span class="st">&quot;h&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## <code>geom_smooth()</code> using method = ‘gam’ and formula ‘y ~ s(x, bs = “cs”)’<br />
## <code>geom_smooth()</code> using method = ‘gam’ and formula ‘y ~ s(x, bs = “cs”)’</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-CEBPB-Combined2_1.png" width="800" height="400" /></p>
</div>
<div id="pseudo-time-heatmaps" class="section level3" number="16.1.2">
<h3><span class="header-section-number">16.1.2</span> Pseudo-time heatmaps</h3>
<p>We can visualize changes in many features across pseudo-time using heatmaps. To do this, we first retrieve the trajectory of interest from the <code>ArchRProject</code> using the <code>getTrajectory()</code> function which returns the trajectory as a <code>SummarizedExperiment</code> object. We will create these pseudo-time heatmaps for motifs, gene scores, gene expression, and peak accessibility by passing the corresponding matrix to the <code>useMatrix</code> parameter.</p>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb338-1"></a>trajMM  &lt;-<span class="st"> </span><span class="kw">getTrajectory</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">name =</span> <span class="st">&quot;MyeloidU&quot;</span>, <span class="dt">useMatrix =</span> <span class="st">&quot;MotifMatrix&quot;</span>, <span class="dt">log2Norm =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## Creating Trajectory Group Matrix..<br />
## Some values are below 0, this could be a DeviationsMatrix in which scaleTo should be set = NULL.<br />
## Continuing without depth normalization!<br />
## Smoothing…</p>
</blockquote>
<p>We then pass this <code>SummarizedExperiment</code> to the <code>plotTrajectoryHeatmap()</code> function.</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb339-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotTrajectoryHeatmap</span>(trajMM, <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;solarExtra&quot;</span>))</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131c8ea2de-Date-2020-04-15_Time-15-01-31.log<br />
## If there is an issue, please report to github with logFile!<br />
## useSeqnames is NULL or greater than 1 with a Sparse.Assays.Matrix trajectory input.<br />
## 2020-04-15 15:01:31 :<br />
## force=FALSE thus continuing with subsetting useSeqnames = z<br />
## 2020-04-15 15:01:31 :<br />
## Removing rows with NA values…<br />
## 2020-04-15 15:01:31 :<br />
## Preparing Main Heatmap..<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131c8ea2de-Date-2020-04-15_Time-15-01-31.log</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Plot-MyeloidU-Traj-Heatmaps_1.png" width="600" height="800" /></p>
<p>We can perform the same steps to obtain a pseudo-time heatmap of gene scores by setting <code>useMatrix = "GeneScoreMatrix"</code>.</p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb340-1"></a>trajGSM &lt;-<span class="st"> </span><span class="kw">getTrajectory</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">name =</span> <span class="st">&quot;MyeloidU&quot;</span>, <span class="dt">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>, <span class="dt">log2Norm =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<blockquote>
<p>## Creating Trajectory Group Matrix..<br />
## Smoothing…</p>
</blockquote>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb341-1"></a>p2 &lt;-<span class="st"> </span><span class="kw">trajectoryHeatmap</span>(trajGSM,  <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;horizonExtra&quot;</span>))</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-trajectoryHeatmap-15e132f1f4173-Date-2020-04-15_Time-15-01-51.log<br />
## If there is an issue, please report to github with logFile!<br />
## Preparing Main Heatmap..<br />
## ArchR logging successful to : ArchRLogs/ArchR-trajectoryHeatmap-15e132f1f4173-Date-2020-04-15_Time-15-01-51.log</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Plot-MyeloidU-Traj-Heatmaps_2.png" width="600" height="800" /></p>
<p>Similarly, we can obtain a pseudo-time heatmap of gene expression by setting <code>useMatrix = "GeneIntegrationMatrix"</code>.</p>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb342-1"></a>trajGIM &lt;-<span class="st"> </span><span class="kw">getTrajectory</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">name =</span> <span class="st">&quot;MyeloidU&quot;</span>, <span class="dt">useMatrix =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>, <span class="dt">log2Norm =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## Creating Trajectory Group Matrix..<br />
## Smoothing…</p>
</blockquote>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb343-1"></a>p3 &lt;-<span class="st"> </span><span class="kw">plotTrajectoryHeatmap</span>(trajGIM,  <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;blueYellow&quot;</span>))</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1333158a4-Date-2020-04-15_Time-15-02-18.log<br />
## If there is an issue, please report to github with logFile!<br />
## Preparing Main Heatmap..<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1333158a4-Date-2020-04-15_Time-15-02-18.log</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Plot-MyeloidU-Traj-Heatmaps_3.png" width="600" height="800" /></p>
<p>Lastly, we can obtain a pseudo-time heatmap of peak accessibility by setting <code>useMatrix = "PeakMatrix"</code>.</p>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb344-1"></a>trajPM  &lt;-<span class="st"> </span><span class="kw">getTrajectory</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">name =</span> <span class="st">&quot;MyeloidU&quot;</span>, <span class="dt">useMatrix =</span> <span class="st">&quot;PeakMatrix&quot;</span>, <span class="dt">log2Norm =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<blockquote>
<p>## Creating Trajectory Group Matrix..<br />
## Smoothing…</p>
</blockquote>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb345-1"></a>p4 &lt;-<span class="st"> </span><span class="kw">plotTrajectoryHeatmap</span>(trajPM, <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;solarExtra&quot;</span>))</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131ecfb872-Date-2020-04-15_Time-15-03-01.log<br />
## If there is an issue, please report to github with logFile!<br />
## Preparing Main Heatmap..<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131ecfb872-Date-2020-04-15_Time-15-03-01.log</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Plot-MyeloidU-Traj-Heatmaps_4.png" width="600" height="800" /></p>
<p>To save an editable vectorized version of this plot, we use <code>plotPDF()</code>.</p>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb346-1"></a><span class="kw">plotPDF</span>(p1, p2, p3, p4, <span class="dt">name =</span> <span class="st">&quot;Plot-MyeloidU-Traj-Heatmaps.pdf&quot;</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">8</span>)</span></code></pre></div>
<blockquote>
<p>## [1] 0</p>
</blockquote>
</div>
<div id="integrative-pseudo-time-analyses" class="section level3" number="16.1.3">
<h3><span class="header-section-number">16.1.3</span> Integrative pseudo-time analyses</h3>
<p>We can also perform integrative analyses, such as identification of positive TF regulators by integration of gene scores / gene expression with motif accessibility across pseudo-time. This can be very powerful, for example in identifying drivers of differentiation. To do this, we use the <code>correlateTrajectories()</code> function which takes two <code>SummarizedExperiment</code> objects retrived from the <code>getTrajectories()</code> function.</p>
<p>First, lets find motifs whose accessibility across pseudo-time is correlated with the gene score of the TF gene.</p>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb347-1"></a>corGSM_MM &lt;-<span class="st"> </span><span class="kw">correlateTrajectories</span>(trajGSM, trajMM)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-correlateTrajectories-15e13b9ca02a-Date-2020-04-15_Time-15-03-37.log<br />
## If there is an issue, please report to github with logFile!<br />
## Found 36 Correlated Pairings!<br />
## 2020-04-15 15:03:56 :</p>
</blockquote>
<p>The primary output of <code>correlateTrajectories()</code> is a list object containing a <code>DataFrame</code> object as the first entry in the list. This <code>DataFrame</code> has columns named <code>idx1</code>, <code>matchname1</code>, <code>name1</code>, and <code>VarAssay1</code> which correspond to the index, match name, unaltered name, and the variance quantile of the features from the first trajectory (gene scores) passed to the <code>correlateTrajectories()</code> function. A “variance quantile” is a normalized measure of the given feature which allows us to derive a correlation across disparate assays. This <code>DataFrame</code> contains all of the features that met the cutoffs specified in the <code>correlateTrajectories()</code> function.</p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb348-1"></a>corGSM_MM[[<span class="dv">1</span>]]</span></code></pre></div>
<blockquote>
<p>## DataFrame with 36 rows and 12 columns<br />
## idx1 idx2 matchname1 matchname2 name1 name2<br />
## <integer> <integer> <array> <array> <character> <character><br />
## 1 82 1081 PRDM16 PRDM16 chr1:PRDM16 z:PRDM16_211<br />
## 2 731 932 TAL1 TAL1 chr1:TAL1 z:TAL1_62<br />
## 3 818 1013 JUN JUN chr1:JUN z:JUN_143<br />
## 4 2034 1002 ATF3 ATF3 chr1:ATF3 z:ATF3_132<br />
## 5 2369 1254 GATA3 GATA3 chr10:GATA3 z:GATA3_384<br />
## … … … … … … …<br />
## 32 20775 1022 CEBPD CEBPD chr8:CEBPD z:CEBPD_152<br />
## 33 20780 1031 SNAI2 SNAI2 chr8:SNAI2 z:SNAI2_161<br />
## 34 21658 1003 NFIL3 NFIL3 chr9:NFIL3 z:NFIL3_133<br />
## 35 21793 1078 KLF4 KLF4 chr9:KLF4 z:KLF4_208<br />
## 36 22097 1565 RXRA RXRA chr9:RXRA z:RXRA_695<br />
## Correlation VarAssay1 VarAssay2 TStat<br />
## <numeric> <numeric> <numeric> <numeric><br />
## 1 0.859588836579798 0.999783802481948 0.860344827586207 16.6530781620713<br />
## 2 0.687938524545333 0.866735849872443 0.987931034482759 9.38348743943718<br />
## 3 0.530335586469274 0.993989708998141 0.963218390804598 6.19265152142139<br />
## 4 0.776280918431044 0.985168850261599 0.936781609195402 12.1905867620732<br />
## 5 0.714559116657296 0.801573917931422 0.989655172413793 10.111530171907<br />
## … … … … …<br />
## 32 0.73809434207349 0.993816750983699 0.998275862068966 10.8296844080467<br />
## 33 0.643337129188769 0.87252994335625 0.983333333333333 8.31877485229719<br />
## 34 0.833740261914099 0.992303368357331 0.893103448275862 14.9479403447508<br />
## 35 0.82265894955927 0.998702814891685 0.905747126436782 14.3243696597426<br />
## 36 0.59509316489084 0.88640982401522 0.890229885057471 7.33039570155334<br />
## Pval FDR<br />
## <numeric> <numeric><br />
## 1 2.47466579579935e-30 4.04855324192774e-27<br />
## 2 2.63756432350363e-15 1.00350121703533e-13<br />
## 3 1.38781951660218e-08 1.55511830764464e-07<br />
## 4 2.36537648993666e-21 3.22479661461365e-19<br />
## 5 6.9322610846337e-17 3.78039304482025e-15<br />
## … … …<br />
## 32 1.93304822797782e-18 1.31769454207155e-16<br />
## 33 5.27319420246801e-13 1.3914428572964e-11<br />
## 34 5.06319064599113e-27 2.70186610322912e-24<br />
## 35 8.9984663184347e-26 2.45358181615986e-23<br />
## 36 6.60626612679289e-11 1.25672690505037e-09</p>
</blockquote>
<p>We can then subset our corresponding trajectory <code>SummarizedExperiment</code> objects to only contain the elements that passed significance above.</p>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb349-1"></a>trajGSM2 &lt;-<span class="st"> </span>trajGSM[corGSM_MM[[<span class="dv">1</span>]]<span class="op">$</span>name1, ]</span>
<span id="cb349-2"><a href="myeloid-trajectory-monocyte-differentiation.html#cb349-2"></a>trajMM2 &lt;-<span class="st"> </span>trajMM[corGSM_MM[[<span class="dv">1</span>]]<span class="op">$</span>name2, ]</span></code></pre></div>
<p>To best order these features, we can create a new trajectory where the values of these two trajectories are multiplied. This will allow us to create side-by-side heatmaps that are identically ordered by row.</p>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb350-1"></a>trajCombined &lt;-<span class="st"> </span>trajGSM2</span>
<span id="cb350-2"><a href="myeloid-trajectory-monocyte-differentiation.html#cb350-2"></a><span class="kw">assay</span>(trajCombined) &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(<span class="kw">assay</span>(trajGSM2), <span class="dv">1</span>, scale)) <span class="op">+</span><span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(<span class="kw">assay</span>(trajMM2), <span class="dv">1</span>, scale))</span></code></pre></div>
<p>We can extract the optimal row order from the return of the <code>plotTrajectoryHeatmap()</code> function.</p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb351-1"></a>combinedMat &lt;-<span class="st"> </span><span class="kw">plotTrajectoryHeatmap</span>(trajCombined, <span class="dt">returnMat =</span> <span class="ot">TRUE</span>, <span class="dt">varCutOff =</span> <span class="dv">0</span>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e136ff55add-Date-2020-04-15_Time-15-03-56.log<br />
## If there is an issue, please report to github with logFile!<br />
## Preparing Main Heatmap..<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e136ff55add-Date-2020-04-15_Time-15-03-56.log</p>
</blockquote>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb352-1"></a>rowOrder &lt;-<span class="st"> </span><span class="kw">match</span>(<span class="kw">rownames</span>(combinedMat), <span class="kw">rownames</span>(trajGSM2))</span></code></pre></div>
<p>With this, we are now ready to create our paired heatmaps. First, we will create the heatmap for the gene score trajectory. We specify the desired row order via the <code>rowOrder</code> parameter.</p>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb353-1"></a>ht1 &lt;-<span class="st"> </span><span class="kw">plotTrajectoryHeatmap</span>(trajGSM2,  <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;horizonExtra&quot;</span>),  <span class="dt">varCutOff =</span> <span class="dv">0</span>, <span class="dt">rowOrder =</span> rowOrder)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e13aac5f3b-Date-2020-04-15_Time-15-03-57.log<br />
## If there is an issue, please report to github with logFile!<br />
## Preparing Main Heatmap..<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e13aac5f3b-Date-2020-04-15_Time-15-03-57.log</p>
</blockquote>
<p>Then, we will create the heatmap for the motif trajectory, again specifying the row order via the <code>rowOrder</code> parameter</p>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb354-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb354-1"></a>ht2 &lt;-<span class="st"> </span><span class="kw">plotTrajectoryHeatmap</span>(trajMM2, <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;solarExtra&quot;</span>), <span class="dt">varCutOff =</span> <span class="dv">0</span>, <span class="dt">rowOrder =</span> rowOrder)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1323caf784-Date-2020-04-15_Time-15-03-57.log<br />
## If there is an issue, please report to github with logFile!<br />
## useSeqnames is NULL or greater than 1 with a Sparse.Assays.Matrix trajectory input.<br />
## 2020-04-15 15:03:58 :<br />
## force=FALSE thus continuing with subsetting useSeqnames = z<br />
## 2020-04-15 15:03:58 :<br />
## Preparing Main Heatmap..<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1323caf784-Date-2020-04-15_Time-15-03-57.log</p>
</blockquote>
<p>Plotting these two heatmaps side-by-side, we see that the rows are matched across the two heatmaps. You may notice that the analysis captures both GATA3 and GATA3-AS1 (an anti-sense transcript of GATA3). This is due to how feature matching is performed and the anti-sense transcript entry could be removed manually in post-processing or programatically if desired.</p>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb355-1"></a>ht1 <span class="op">+</span><span class="st"> </span>ht2</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-MyeloidU-GS-TF_1.png" width="800" height="600" /></p>
<p>We can repeat this same exact process but use gene expression from the <code>GeneIntegrationMatrix</code> instead of gene scores. Because this is the same analytical workflow, we do not repeat our explanations for each step.</p>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb356-1"></a>corGIM_MM &lt;-<span class="st"> </span><span class="kw">correlateTrajectories</span>(trajGIM, trajMM)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-correlateTrajectories-15e13f7dfd04-Date-2020-04-15_Time-15-04-14.log<br />
## If there is an issue, please report to github with logFile!<br />
## Found 54 Correlated Pairings!<br />
## 2020-04-15 15:04:30 :</p>
</blockquote>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb357-1"></a>corGIM_MM[[<span class="dv">1</span>]]</span></code></pre></div>
<blockquote>
<p>## DataFrame with 54 rows and 12 columns<br />
## idx1 idx2 matchname1 matchname2 name1 name2<br />
## <integer> <integer> <array> <array> <character> <character><br />
## 1 295 1601 RUNX3 RUNX3 chr1:RUNX3 z:RUNX3_731<br />
## 2 680 1612 NFIA NFIA chr1:NFIA z:NFIA_742<br />
## 3 1227 1512 MEF2D MEF2D chr1:MEF2D z:MEF2D_642<br />
## 4 1317 930 USF1 USF1 chr1:USF1 z:USF1_60<br />
## 5 1664 1002 ATF3 ATF3 chr1:ATF3 z:ATF3_132<br />
## … … … … … … …<br />
## 50 16981 1696 KLF10 KLF10 chr8:KLF10 z:KLF10_826<br />
## 51 17463 1003 NFIL3 NFIL3 chr9:NFIL3 z:NFIL3_133<br />
## 52 17560 1078 KLF4 KLF4 chr9:KLF4 z:KLF4_208<br />
## 53 17803 1565 RXRA RXRA chr9:RXRA z:RXRA_695<br />
## 54 18090 888 TFE3 TFE3 chrX:TFE3 z:TFE3_18<br />
## Correlation VarAssay1 VarAssay2 TStat<br />
## <numeric> <numeric> <numeric> <numeric><br />
## 1 0.506721180509532 0.871189720982743 0.98448275862069 5.81861222342308<br />
## 2 0.899884055575892 0.878017310897264 0.919540229885057 20.4260514493749<br />
## 3 0.787682536829653 0.815493790656416 0.841379310344828 12.656866457322<br />
## 4 0.653303634000368 0.804204075049728 0.875287356321839 8.54234597517229<br />
## 5 0.847460093565636 0.813020805333047 0.936781609195402 15.8034906427774<br />
## … … … … …<br />
## 50 0.57964654735637 0.953335842159024 0.864367816091954 7.04188649257934<br />
## 51 0.849563937747037 0.933928283425622 0.893103448275862 15.9440428922802<br />
## 52 0.820699669277503 0.98876404494382 0.905747126436782 14.2196070232556<br />
## 53 0.899871735441299 0.960485995376593 0.890229885057471 20.424581355724<br />
## 54 0.832322259515336 0.801247244771786 0.886206896551724 14.8650341763864<br />
## Pval FDR<br />
## <numeric> <numeric><br />
## 1 7.47838268446286e-08 2.00835886256963e-07<br />
## 2 4.28849849649256e-37 2.40519926139511e-35<br />
## 3 2.45911764921563e-22 2.0096926995314e-21<br />
## 4 1.7427906036477e-13 7.90509804908143e-13<br />
## 5 1.05531067509617e-28 1.5631789374862e-27<br />
## … … …<br />
## 50 2.63272438071192e-10 8.66605108651006e-10<br />
## 51 5.63648466586524e-29 9.05658892074619e-28<br />
## 52 1.46590261078753e-25 1.5687313439583e-24<br />
## 53 4.31312103836676e-37 2.40519926139511e-35<br />
## 54 7.4030081939147e-27 8.88360983269764e-26</p>
</blockquote>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb358-1"></a>trajGIM2 &lt;-<span class="st"> </span>trajGIM[corGIM_MM[[<span class="dv">1</span>]]<span class="op">$</span>name1, ]</span>
<span id="cb358-2"><a href="myeloid-trajectory-monocyte-differentiation.html#cb358-2"></a>trajMM2 &lt;-<span class="st"> </span>trajMM[corGIM_MM[[<span class="dv">1</span>]]<span class="op">$</span>name2, ]</span>
<span id="cb358-3"><a href="myeloid-trajectory-monocyte-differentiation.html#cb358-3"></a></span>
<span id="cb358-4"><a href="myeloid-trajectory-monocyte-differentiation.html#cb358-4"></a>trajCombined &lt;-<span class="st"> </span>trajGIM2</span>
<span id="cb358-5"><a href="myeloid-trajectory-monocyte-differentiation.html#cb358-5"></a><span class="kw">assay</span>(trajCombined) &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(<span class="kw">assay</span>(trajGIM2), <span class="dv">1</span>, scale)) <span class="op">+</span><span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(<span class="kw">assay</span>(trajMM2), <span class="dv">1</span>, scale))</span>
<span id="cb358-6"><a href="myeloid-trajectory-monocyte-differentiation.html#cb358-6"></a></span>
<span id="cb358-7"><a href="myeloid-trajectory-monocyte-differentiation.html#cb358-7"></a>combinedMat &lt;-<span class="st"> </span><span class="kw">plotTrajectoryHeatmap</span>(trajCombined, <span class="dt">returnMat =</span> <span class="ot">TRUE</span>, <span class="dt">varCutOff =</span> <span class="dv">0</span>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e137bbcc972-Date-2020-04-15_Time-15-04-30.log<br />
## If there is an issue, please report to github with logFile!<br />
## Preparing Main Heatmap..<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e137bbcc972-Date-2020-04-15_Time-15-04-30.log</p>
</blockquote>
<div class="sourceCode" id="cb359"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb359-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb359-1"></a>rowOrder &lt;-<span class="st"> </span><span class="kw">match</span>(<span class="kw">rownames</span>(combinedMat), <span class="kw">rownames</span>(trajGIM2))</span></code></pre></div>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb360-1"></a>ht1 &lt;-<span class="st"> </span><span class="kw">plotTrajectoryHeatmap</span>(trajGIM2,  <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;blueYellow&quot;</span>),  <span class="dt">varCutOff =</span> <span class="dv">0</span>, <span class="dt">rowOrder =</span> rowOrder)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1331e06f29-Date-2020-04-15_Time-15-04-31.log<br />
## If there is an issue, please report to github with logFile!<br />
## Preparing Main Heatmap..<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1331e06f29-Date-2020-04-15_Time-15-04-31.log</p>
</blockquote>
<div class="sourceCode" id="cb361"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb361-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb361-1"></a>ht2 &lt;-<span class="st"> </span><span class="kw">plotTrajectoryHeatmap</span>(trajMM2, <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;solarExtra&quot;</span>), <span class="dt">varCutOff =</span> <span class="dv">0</span>, <span class="dt">rowOrder =</span> rowOrder)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1345e2da25-Date-2020-04-15_Time-15-04-32.log<br />
## If there is an issue, please report to github with logFile!<br />
## useSeqnames is NULL or greater than 1 with a Sparse.Assays.Matrix trajectory input.<br />
## 2020-04-15 15:04:32 :<br />
## force=FALSE thus continuing with subsetting useSeqnames = z<br />
## 2020-04-15 15:04:32 :<br />
## Preparing Main Heatmap..<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1345e2da25-Date-2020-04-15_Time-15-04-32.log</p>
</blockquote>
<div class="sourceCode" id="cb362"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb362-1"><a href="myeloid-trajectory-monocyte-differentiation.html#cb362-1"></a>ht1 <span class="op">+</span><span class="st"> </span>ht2</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-MyeloidU-GEx-TF_1.png" width="800" height="600" /></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="trajectory-analysis-with-archr.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="lymphoid-trajectory-b-cell-cifferentiation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/16_Trajectory.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/16_Trajectory.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
},
"search": true,
"info": false
});
});
</script>

</body>

</html>
