<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>15.3 Peak2GeneLinkage with ArchR | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="15.3 Peak2GeneLinkage with ArchR | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="15.3 Peak2GeneLinkage with ArchR | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-05-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="co-accessibility-with-archr.html"/>
<link rel="next" href="identification-of-positive-tf-regulators.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://www.archrproject.com/index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li><a href="index.html#section"><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>1.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="1.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>1.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="1.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>1.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>1.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="1.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="1.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>1.7</b> Per-cell Quality Control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>2.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="2.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="3.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>4.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="4.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>4.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="4.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>5.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>6.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="6.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>6.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="6.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>6.3</b> Dimensionality Reduction After Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>7</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>7.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="7.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>7.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="7.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>7.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="7.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>7.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="7.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="7.6" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.6</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.7" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.7</b> Launching the ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>8</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="8.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>8.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>9</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="9.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>10.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="10.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>10.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="10.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>10.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="10.4" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>10.4</b> Add Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>11</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>11.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="11.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>11.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="11.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>11.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-enrichment-in-differential-peaks.html"><a href="motif-enrichment-in-differential-peaks.html"><i class="fa fa-check"></i><b>12.1</b> Motif Enrichment in Differential Peaks</a></li>
<li class="chapter" data-level="12.2" data-path="motif-enrichment-in-marker-peaks.html"><a href="motif-enrichment-in-marker-peaks.html"><i class="fa fa-check"></i><b>12.2</b> Motif Enrichment in Marker Peaks</a></li>
<li class="chapter" data-level="12.3" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>12.3</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="12.4" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>12.4</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>13</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>13.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="13.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>13.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>14</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>14.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="14.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>14.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="14.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>14.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>15</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>15.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="15.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>15.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="15.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>15.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="15.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>15.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>16</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="myeloid-trajectory-monocyte-differentiation.html"><a href="myeloid-trajectory-monocyte-differentiation.html"><i class="fa fa-check"></i><b>16.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="16.2" data-path="lymphoid-trajectory-b-cell-cifferentiation.html"><a href="lymphoid-trajectory-b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>16.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>17</b> Plot aesthetics in ArchR</a></li>
<li class="chapter" data-level="18" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>18</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="index.html">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="peak2genelinkage-with-archr" class="section level2" number="15.3">
<h2><span class="header-section-number">15.3</span> Peak2GeneLinkage with ArchR</h2>
<p>Similar to co-accessibility, ArchR can also identify so-called “peak-to-gene links”. The primary differences between peak-to-gene links and co-accessibility is that co-accessibility is an ATAC-seq-only analysis that looks for correlations in accessibility between two peaks while peak-to-gene linkage leverages integrated scRNA-seq data to look for correlations between peak accessibility and gene expression. These represent orthogonal approaches to a similar problem. However, because peak-to-gene linkage correlates scATAC-seq and scRNA-seq data, we often think of these links as more relevant to gene regulatory interactions.</p>
<p>To identify peak-to-gene links in ArchR, we use the <code>addPeak2GeneLinks()</code> function.</p>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="peak2genelinkage-with-archr.html#cb299-1"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addPeak2GeneLinks</span>(</span>
<span id="cb299-2"><a href="peak2genelinkage-with-archr.html#cb299-2"></a>    <span class="dt">ArchRProj =</span> projHeme5,</span>
<span id="cb299-3"><a href="peak2genelinkage-with-archr.html#cb299-3"></a>    <span class="dt">reducedDims =</span> <span class="st">&quot;IterativeLSI&quot;</span></span>
<span id="cb299-4"><a href="peak2genelinkage-with-archr.html#cb299-4"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-addPeak2GeneLinks-11a7017ae41d-Date-2020-04-15_Time-11-57-10.log<br />
## If there is an issue, please report to github with logFile!<br />
## 2020-04-15 11:57:11 : Getting Available Matrices, 0.006 mins elapsed.<br />
## 2020-04-15 11:57:11 : Filtered Low Prediction Score Cells (1049 of 10251, 0.102), 0.004 mins elapsed.<br />
## 2020-04-15 11:57:12 : Computing KNN, 0.009 mins elapsed.<br />
## 2020-04-15 11:57:12 : Identifying Non-Overlapping KNN pairs, 0.016 mins elapsed.<br />
## 2020-04-15 11:57:15 : Identified 491 Groupings!, 0.07 mins elapsed.<br />
## 2020-04-15 11:57:15 : Getting Group RNA Matrix, 0.071 mins elapsed.<br />
## 2020-04-15 11:57:43 : Getting Group ATAC Matrix, 0.536 mins elapsed.<br />
## 2020-04-15 11:58:18 : Normalizing Group Matrices, 1.112 mins elapsed.<br />
## 2020-04-15 11:58:24 : Finding Peak Gene Pairings, 1.218 mins elapsed.<br />
## 2020-04-15 11:58:25 : Computing Correlations, 1.227 mins elapsed.<br />
## 2020-04-15 11:58:34 : Completed Peak2Gene Correlations!, 1.382 mins elapsed.<br />
## ArchR logging successful to : ArchRLogs/ArchR-addPeak2GeneLinks-11a7017ae41d-Date-2020-04-15_Time-11-57-10.log</p>
</blockquote>
<p>We can then retrieve these peak-to-gene links in a similar fashion to how we retrieved co-accessibility interactions by using the <code>getPeak2GeneLinks()</code> function. As we saw previously, this function allows for a user-specified cutoff for correlation and resolution for linkages.</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="peak2genelinkage-with-archr.html#cb300-1"></a>p2g &lt;-<span class="st"> </span><span class="kw">getPeak2GeneLinks</span>(</span>
<span id="cb300-2"><a href="peak2genelinkage-with-archr.html#cb300-2"></a>    <span class="dt">ArchRProj =</span> projHeme5,</span>
<span id="cb300-3"><a href="peak2genelinkage-with-archr.html#cb300-3"></a>    <span class="dt">corCutOff =</span> <span class="fl">0.45</span>,</span>
<span id="cb300-4"><a href="peak2genelinkage-with-archr.html#cb300-4"></a>    <span class="dt">resolution =</span> <span class="dv">1</span>,</span>
<span id="cb300-5"><a href="peak2genelinkage-with-archr.html#cb300-5"></a>    <span class="dt">returnLoops =</span> <span class="ot">FALSE</span></span>
<span id="cb300-6"><a href="peak2genelinkage-with-archr.html#cb300-6"></a>)</span></code></pre></div>
<p>When <code>returnLoops</code> is set to false, this function returns a <code>DataFrame</code> object anaolgous to the <code>DataFrame</code> object returned by <code>getCoAccessibility()</code>. The primary difference is that the indexes for the scATAC-seq peaks are stored in the <code>idxATAC</code> column and the indexes for the scRNA-seq gene are stored in the <code>idxRNA</code> column.</p>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="peak2genelinkage-with-archr.html#cb301-1"></a>p2g</span></code></pre></div>
<blockquote>
<p>## DataFrame with 43754 rows and 6 columns<br />
## idxATAC idxRNA Correlation FDR<br />
## <integer> <integer> <numeric> <numeric><br />
## 1 47 5 0.549552663393716 1.34094093110629e-38<br />
## 2 3 6 0.487418258348982 2.1460798658766e-29<br />
## 3 3 7 0.821369314290254 1.70141157082175e-118<br />
## 4 47 7 0.569899085935183 4.61958766180263e-42<br />
## 5 59 7 0.549954043409585 1.15152278593831e-38<br />
## … … … … …<br />
## 43750 143964 18590 0.520703047216992 4.37554114264876e-34<br />
## 43751 143977 18590 0.546301068655939 4.54936701954597e-38<br />
## 43752 144004 18594 0.47934214168793 2.47516776950818e-28<br />
## 43753 143999 18598 0.49092225993235 7.28579053551407e-30<br />
## 43754 144005 18598 0.493389088498317 3.37643947034054e-30<br />
## VarQATAC VarQRNA<br />
## <numeric> <numeric><br />
## 1 0.948753202924817 0.793290683296597<br />
## 2 0.253206396822421 0.445890005913661<br />
## 3 0.253206396822421 0.42728885543788<br />
## 4 0.948753202924817 0.42728885543788<br />
## 5 0.916137185870328 0.42728885543788<br />
## … … …<br />
## 43750 0.629044018082203 0.960701037578625<br />
## 43751 0.701518655084057 0.960701037578625<br />
## 43752 0.814587977140318 0.802537497983979<br />
## 43753 0.0392197709865356 0.901295629267244<br />
## 43754 0.158948399058392 0.901295629267244</p>
</blockquote>
<p>This peak-to-gene linkage <code>DataFrame</code> also has a metadata component containing a <code>GRanges</code> object of the relevant peaks. The indexes of <code>idxATAC</code> mentioned above apply to this <code>GRanges</code> object.</p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="peak2genelinkage-with-archr.html#cb302-1"></a><span class="kw">metadata</span>(p2g)[[<span class="dv">1</span>]]</span></code></pre></div>
<blockquote>
<p>## GRanges object with 144009 ranges and 0 metadata columns:
## seqnames ranges strand<br />
## <Rle> <IRanges> <Rle><br />
## [1] chr1 752499-752999 *<br />
## [2] chr1 762651-763151 *<br />
## [3] chr1 801006-801506 *<br />
## [4] chr1 805039-805539 *<br />
## [5] chr1 845325-845825 *<br />
## … … … …<br />
## [144005] chrX 154664540-154665040 *<br />
## [144006] chrX 154807324-154807824 *<br />
## [144007] chrX 154840785-154841285 *<br />
## [144008] chrX 154842404-154842904 *<br />
## [144009] chrX 154862017-154862517 *<br />
## ——-<br />
## seqinfo: 23 sequences from an unspecified genome; no seqlengths</p>
</blockquote>
<p>If we set <code>returnLoops = TRUE</code>, then <code>getPeak2GeneLinks()</code> will return a loop track <code>GRanges</code> object that connects the peak and gene. As for co-accessibility, the start and end of the <code>IRanges</code> object represent the position of the peak and gene being linked. When <code>resolution = 1</code>, this links the center of the peak to the single-base TSS of the gene.</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="peak2genelinkage-with-archr.html#cb303-1"></a>p2g &lt;-<span class="st"> </span><span class="kw">getPeak2GeneLinks</span>(</span>
<span id="cb303-2"><a href="peak2genelinkage-with-archr.html#cb303-2"></a>    <span class="dt">ArchRProj =</span> projHeme5,</span>
<span id="cb303-3"><a href="peak2genelinkage-with-archr.html#cb303-3"></a>    <span class="dt">corCutOff =</span> <span class="fl">0.45</span>,</span>
<span id="cb303-4"><a href="peak2genelinkage-with-archr.html#cb303-4"></a>    <span class="dt">resolution =</span> <span class="dv">1</span>,</span>
<span id="cb303-5"><a href="peak2genelinkage-with-archr.html#cb303-5"></a>    <span class="dt">returnLoops =</span> <span class="ot">TRUE</span></span>
<span id="cb303-6"><a href="peak2genelinkage-with-archr.html#cb303-6"></a>)</span>
<span id="cb303-7"><a href="peak2genelinkage-with-archr.html#cb303-7"></a></span>
<span id="cb303-8"><a href="peak2genelinkage-with-archr.html#cb303-8"></a>p2g[[<span class="dv">1</span>]]</span></code></pre></div>
<blockquote>
<p>## GRanges object with 43695 ranges and 2 metadata columns:
## seqnames ranges strand | value<br />
## <Rle> <IRanges> <Rle> | <numeric><br />
## [1] chr1 762901-948847 * | 0.533350285896763<br />
## [2] chr1 801256-895967 * | 0.487418258348982<br />
## [3] chr1 801256-901877 * | 0.821369314290254<br />
## [4] chr1 852367-955503 * | 0.487693663181145<br />
## [5] chr1 894679-968584 * | 0.549552663393716<br />
## … … … … . …<br />
## [43691] chrX 153962451-153979858 * | 0.453720046871409<br />
## [43692] chrX 153980218-153991031 * | 0.546301068655939<br />
## [43693] chrX 154255064-154493795 * | 0.47934214168793<br />
## [43694] chrX 154356425-154444701 * | 0.49092225993235<br />
## [43695] chrX 154444701-154664790 * | 0.493389088498317<br />
## FDR<br />
## <numeric><br />
## [1] 5.20045729958651e-36<br />
## [2] 2.1460798658766e-29<br />
## [3] 1.70141157082175e-118<br />
## [4] 1.97244775926907e-29<br />
## [5] 1.34094093110629e-38<br />
## … …<br />
## [43691] 3.78754241443136e-25<br />
## [43692] 4.54936701954597e-38<br />
## [43693] 2.47516776950818e-28<br />
## [43694] 7.28579053551407e-30<br />
## [43695] 3.37643947034054e-30<br />
## ——-<br />
## seqinfo: 23 sequences from an unspecified genome; no seqlengths</p>
</blockquote>
<p>We can alternatively decrease the resolution of these links by setting <code>resolution = 1000</code>. This is primarily useful for plotting the links as a browser tracks because there are instances where many nearby peaks all link to the same gene and this can be difficult to visualize.</p>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="peak2genelinkage-with-archr.html#cb304-1"></a>p2g &lt;-<span class="st"> </span><span class="kw">getPeak2GeneLinks</span>(</span>
<span id="cb304-2"><a href="peak2genelinkage-with-archr.html#cb304-2"></a>    <span class="dt">ArchRProj =</span> projHeme5,</span>
<span id="cb304-3"><a href="peak2genelinkage-with-archr.html#cb304-3"></a>    <span class="dt">corCutOff =</span> <span class="fl">0.45</span>,</span>
<span id="cb304-4"><a href="peak2genelinkage-with-archr.html#cb304-4"></a>    <span class="dt">resolution =</span> <span class="dv">1000</span>,</span>
<span id="cb304-5"><a href="peak2genelinkage-with-archr.html#cb304-5"></a>    <span class="dt">returnLoops =</span> <span class="ot">TRUE</span></span>
<span id="cb304-6"><a href="peak2genelinkage-with-archr.html#cb304-6"></a>)</span>
<span id="cb304-7"><a href="peak2genelinkage-with-archr.html#cb304-7"></a></span>
<span id="cb304-8"><a href="peak2genelinkage-with-archr.html#cb304-8"></a>p2g[[<span class="dv">1</span>]]</span></code></pre></div>
<blockquote>
<p>## GRanges object with 42126 ranges and 2 metadata columns:<br />
## seqnames ranges strand | value<br />
## <Rle> <IRanges> <Rle> | <numeric><br />
## [1] chr1 762500-948500 * | 0.533350285896763<br />
## [2] chr1 801500-895500 * | 0.487418258348982<br />
## [3] chr1 801500-901500 * | 0.821369314290254<br />
## [4] chr1 852500-955500 * | 0.487693663181145<br />
## [5] chr1 894500-968500 * | 0.549552663393716<br />
## … … … … . …<br />
## [42122] chrX 153962500-153979500 * | 0.453720046871409<br />
## [42123] chrX 153980500-153991500 * | 0.546301068655939<br />
## [42124] chrX 154255500-154493500 * | 0.47934214168793<br />
## [42125] chrX 154356500-154444500 * | 0.49092225993235<br />
## [42126] chrX 154444500-154664500 * | 0.493389088498317<br />
## FDR<br />
## <numeric><br />
## [1] 5.20045729958651e-36<br />
## [2] 2.1460798658766e-29<br />
## [3] 1.70141157082175e-118<br />
## [4] 1.97244775926907e-29<br />
## [5] 1.34094093110629e-38<br />
## … …<br />
## [42122] 3.78754241443136e-25<br />
## [42123] 4.54936701954597e-38<br />
## [42124] 2.47516776950818e-28<br />
## [42125] 7.28579053551407e-30<br />
## [42126] 3.37643947034054e-30<br />
## ——-<br />
## seqinfo: 23 sequences from an unspecified genome; no seqlengths</p>
</blockquote>
<p>Decreasing the resolution even further also decreases the total number of peak-to-gene links identified.</p>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="peak2genelinkage-with-archr.html#cb305-1"></a>p2g &lt;-<span class="st"> </span><span class="kw">getPeak2GeneLinks</span>(</span>
<span id="cb305-2"><a href="peak2genelinkage-with-archr.html#cb305-2"></a>    <span class="dt">ArchRProj =</span> projHeme5,</span>
<span id="cb305-3"><a href="peak2genelinkage-with-archr.html#cb305-3"></a>    <span class="dt">corCutOff =</span> <span class="fl">0.45</span>,</span>
<span id="cb305-4"><a href="peak2genelinkage-with-archr.html#cb305-4"></a>    <span class="dt">resolution =</span> <span class="dv">10000</span>,</span>
<span id="cb305-5"><a href="peak2genelinkage-with-archr.html#cb305-5"></a>    <span class="dt">returnLoops =</span> <span class="ot">TRUE</span></span>
<span id="cb305-6"><a href="peak2genelinkage-with-archr.html#cb305-6"></a>)</span>
<span id="cb305-7"><a href="peak2genelinkage-with-archr.html#cb305-7"></a></span>
<span id="cb305-8"><a href="peak2genelinkage-with-archr.html#cb305-8"></a>p2g[[<span class="dv">1</span>]]</span></code></pre></div>
<blockquote>
<p>## GRanges object with 33645 ranges and 2 metadata columns:<br />
## seqnames ranges strand | value<br />
## <Rle> <IRanges> <Rle> | <numeric><br />
## [1] chr1 765000-945000 * | 0.533350285896763<br />
## [2] chr1 805000-895000 * | 0.487418258348982<br />
## [3] chr1 805000-905000 * | 0.821369314290254<br />
## [4] chr1 855000-955000 * | 0.487693663181145<br />
## [5] chr1 895000-965000 * | 0.549552663393716<br />
## … … … … . …<br />
## [33641] chrX 153965000-153975000 * | 0.453720046871409<br />
## [33642] chrX 153985000-153995000 * | 0.546301068655939<br />
## [33643] chrX 154255000-154495000 * | 0.47934214168793<br />
## [33644] chrX 154355000-154445000 * | 0.49092225993235<br />
## [33645] chrX 154445000-154665000 * | 0.493389088498317<br />
## FDR<br />
## <numeric><br />
## [1] 5.20045729958651e-36<br />
## [2] 2.1460798658766e-29<br />
## [3] 1.70141157082175e-118<br />
## [4] 1.97244775926907e-29<br />
## [5] 1.34094093110629e-38<br />
## … …<br />
## [33641] 3.78754241443136e-25<br />
## [33642] 4.54936701954597e-38<br />
## [33643] 2.47516776950818e-28<br />
## [33644] 7.28579053551407e-30<br />
## [33645] 3.37643947034054e-30<br />
## ——-<br />
## seqinfo: 23 sequences from an unspecified genome; no seqlengths</p>
</blockquote>
<div id="plotting-browser-tracks-with-peak-to-gene-links" class="section level3" number="15.3.1">
<h3><span class="header-section-number">15.3.1</span> Plotting browser tracks with peak-to-gene links</h3>
<p>To plot these peak-to-gene links as a browser track, we use the same workflow shown for co-accessibility in the previous section. Here we use the <code>plotBrowserTrack()</code> function</p>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="peak2genelinkage-with-archr.html#cb306-1"></a>markerGenes  &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb306-2"><a href="peak2genelinkage-with-archr.html#cb306-2"></a>    <span class="st">&quot;CD34&quot;</span>, <span class="co">#Early Progenitor</span></span>
<span id="cb306-3"><a href="peak2genelinkage-with-archr.html#cb306-3"></a>    <span class="st">&quot;GATA1&quot;</span>, <span class="co">#Erythroid</span></span>
<span id="cb306-4"><a href="peak2genelinkage-with-archr.html#cb306-4"></a>    <span class="st">&quot;PAX5&quot;</span>, <span class="st">&quot;MS4A1&quot;</span>, <span class="co">#B-Cell Trajectory</span></span>
<span id="cb306-5"><a href="peak2genelinkage-with-archr.html#cb306-5"></a>    <span class="st">&quot;CD14&quot;</span>, <span class="co">#Monocytes</span></span>
<span id="cb306-6"><a href="peak2genelinkage-with-archr.html#cb306-6"></a>    <span class="st">&quot;CD3D&quot;</span>, <span class="st">&quot;CD8A&quot;</span>, <span class="st">&quot;TBX21&quot;</span>, <span class="st">&quot;IL7R&quot;</span> <span class="co">#TCells</span></span>
<span id="cb306-7"><a href="peak2genelinkage-with-archr.html#cb306-7"></a>  )</span>
<span id="cb306-8"><a href="peak2genelinkage-with-archr.html#cb306-8"></a></span>
<span id="cb306-9"><a href="peak2genelinkage-with-archr.html#cb306-9"></a>p &lt;-<span class="st"> </span><span class="kw">plotBrowserTrack</span>(</span>
<span id="cb306-10"><a href="peak2genelinkage-with-archr.html#cb306-10"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb306-11"><a href="peak2genelinkage-with-archr.html#cb306-11"></a>    <span class="dt">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>, </span>
<span id="cb306-12"><a href="peak2genelinkage-with-archr.html#cb306-12"></a>    <span class="dt">geneSymbol =</span> markerGenes, </span>
<span id="cb306-13"><a href="peak2genelinkage-with-archr.html#cb306-13"></a>    <span class="dt">upstream =</span> <span class="dv">50000</span>,</span>
<span id="cb306-14"><a href="peak2genelinkage-with-archr.html#cb306-14"></a>    <span class="dt">downstream =</span> <span class="dv">50000</span>,</span>
<span id="cb306-15"><a href="peak2genelinkage-with-archr.html#cb306-15"></a>    <span class="dt">loops =</span> <span class="kw">getPeak2GeneLinks</span>(projHeme5)</span>
<span id="cb306-16"><a href="peak2genelinkage-with-archr.html#cb306-16"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotBrowserTrack-11a7048fa912a-Date-2020-04-15_Time-11-58-35.log<br />
## If there is an issue, please report to github with logFile!<br />
## 2020-04-15 11:58:36 : Validating Region, 0.012 mins elapsed.<br />
## GRanges object with 9 ranges and 2 metadata columns:<br />
## seqnames ranges strand | gene_id symbol<br />
## <Rle> <IRanges> <Rle> | <character> <character><br />
## [1] chr1 208059883-208084683 - | 947 CD34<br />
## [2] chrX 48644982-48652717 + | 2623 GATA1<br />
## [3] chr9 36838531-37034476 - | 5079 PAX5<br />
## [4] chr11 60223282-60238225 + | 931 MS4A1<br />
## [5] chr5 140011313-140013286 - | 929 CD14<br />
## [6] chr11 118209789-118213459 - | 915 CD3D<br />
## [7] chr2 87011728-87035519 - | 925 CD8A<br />
## [8] chr17 45810610-45823485 + | 30009 TBX21<br />
## [9] chr5 35856977-35879705 + | 3575 IL7R<br />
## ——-<br />
## seqinfo: 24 sequences from hg19 genome<br />
## 2020-04-15 11:58:36 : Adding Bulk Tracks (1 of 9), 0.015 mins elapsed.<br />
## 2020-04-15 11:58:37 : Adding Feature Tracks (1 of 9), 0.038 mins elapsed.<br />
## 2020-04-15 11:58:37 : Adding Loop Tracks (1 of 9), 0.039 mins elapsed.<br />
## 2020-04-15 11:58:37 : Adding Gene Tracks (1 of 9), 0.041 mins elapsed.<br />
## 2020-04-15 11:58:38 : Plotting, 0.045 mins elapsed.<br />
## 2020-04-15 11:58:39 : Adding Bulk Tracks (2 of 9), 0.076 mins elapsed.<br />
## 2020-04-15 11:58:40 : Adding Feature Tracks (2 of 9), 0.091 mins elapsed.<br />
## 2020-04-15 11:58:41 : Adding Loop Tracks (2 of 9), 0.095 mins elapsed.<br />
## 2020-04-15 11:58:41 : Adding Gene Tracks (2 of 9), 0.098 mins elapsed.<br />
## 2020-04-15 11:58:41 : Plotting, 0.104 mins elapsed.<br />
## 2020-04-15 11:58:43 : Adding Bulk Tracks (3 of 9), 0.134 mins elapsed.<br />
## 2020-04-15 11:58:44 : Adding Feature Tracks (3 of 9), 0.151 mins elapsed.<br />
## 2020-04-15 11:58:44 : Adding Loop Tracks (3 of 9), 0.153 mins elapsed.<br />
## 2020-04-15 11:58:44 : Adding Gene Tracks (3 of 9), 0.157 mins elapsed.<br />
## 2020-04-15 11:58:45 : Plotting, 0.162 mins elapsed.<br />
## 2020-04-15 11:58:47 : Adding Bulk Tracks (4 of 9), 0.196 mins elapsed.<br />
## 2020-04-15 11:58:48 : Adding Feature Tracks (4 of 9), 0.212 mins elapsed.<br />
## 2020-04-15 11:58:48 : Adding Loop Tracks (4 of 9), 0.213 mins elapsed.<br />
## 2020-04-15 11:58:48 : Adding Gene Tracks (4 of 9), 0.216 mins elapsed.<br />
## 2020-04-15 11:58:48 : Plotting, 0.22 mins elapsed.<br />
## 2020-04-15 11:58:49 : Adding Bulk Tracks (5 of 9), 0.242 mins elapsed.<br />
## 2020-04-15 11:58:50 : Adding Feature Tracks (5 of 9), 0.258 mins elapsed.<br />
## 2020-04-15 11:58:51 : Adding Loop Tracks (5 of 9), 0.26 mins elapsed.<br />
## 2020-04-15 11:58:51 : Adding Gene Tracks (5 of 9), 0.268 mins elapsed.<br />
## 2020-04-15 11:58:51 : Plotting, 0.273 mins elapsed.<br />
## 2020-04-15 11:58:53 : Adding Bulk Tracks (6 of 9), 0.307 mins elapsed.<br />
## 2020-04-15 11:58:54 : Adding Feature Tracks (6 of 9), 0.323 mins elapsed.<br />
## 2020-04-15 11:58:54 : Adding Loop Tracks (6 of 9), 0.325 mins elapsed.<br />
## 2020-04-15 11:58:55 : Adding Gene Tracks (6 of 9), 0.333 mins elapsed.<br />
## 2020-04-15 11:58:55 : Plotting, 0.338 mins elapsed.<br />
## 2020-04-15 11:58:57 : Adding Bulk Tracks (7 of 9), 0.369 mins elapsed.<br />
## 2020-04-15 11:58:58 : Adding Feature Tracks (7 of 9), 0.389 mins elapsed.<br />
## 2020-04-15 11:58:58 : Adding Loop Tracks (7 of 9), 0.391 mins elapsed.<br />
## 2020-04-15 11:58:59 : Adding Gene Tracks (7 of 9), 0.397 mins elapsed.<br />
## 2020-04-15 11:58:59 : Plotting, 0.403 mins elapsed.<br />
## 2020-04-15 11:59:01 : Adding Bulk Tracks (8 of 9), 0.443 mins elapsed.<br />
## 2020-04-15 11:59:03 : Adding Feature Tracks (8 of 9), 0.462 mins elapsed.<br />
## 2020-04-15 11:59:03 : Adding Loop Tracks (8 of 9), 0.464 mins elapsed.<br />
## 2020-04-15 11:59:03 : Adding Gene Tracks (8 of 9), 0.468 mins elapsed.<br />
## 2020-04-15 11:59:03 : Plotting, 0.473 mins elapsed.<br />
## 2020-04-15 11:59:06 : Adding Bulk Tracks (9 of 9), 0.515 mins elapsed.<br />
## 2020-04-15 11:59:07 : Adding Feature Tracks (9 of 9), 0.533 mins elapsed.<br />
## 2020-04-15 11:59:07 : Adding Loop Tracks (9 of 9), 0.536 mins elapsed.<br />
## 2020-04-15 11:59:07 : Adding Gene Tracks (9 of 9), 0.543 mins elapsed.<br />
## 2020-04-15 11:59:08 : Plotting, 0.549 mins elapsed.<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotBrowserTrack-11a7048fa912a-Date-2020-04-15_Time-11-58-35.log</p>
</blockquote>
<p>To plot our browser track we use the <code>grid.draw</code> function and select a specific marker gene by name using the <code>$</code> accessor.</p>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="peak2genelinkage-with-archr.html#cb307-1"></a>grid<span class="op">::</span><span class="kw">grid.newpage</span>()</span>
<span id="cb307-2"><a href="peak2genelinkage-with-archr.html#cb307-2"></a>grid<span class="op">::</span><span class="kw">grid.draw</span>(p<span class="op">$</span>CD14)</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-Tracks-Marker-Genes-with-Peak2GeneLinks_5.png" width="500" height="500" /></p>
<p>To save an editable vectorized version of this plot, we use <code>plotPDF()</code>.</p>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="peak2genelinkage-with-archr.html#cb308-1"></a><span class="kw">plotPDF</span>(<span class="dt">plotList =</span> p, </span>
<span id="cb308-2"><a href="peak2genelinkage-with-archr.html#cb308-2"></a>    <span class="dt">name =</span> <span class="st">&quot;Plot-Tracks-Marker-Genes-with-Peak2GeneLinks.pdf&quot;</span>, </span>
<span id="cb308-3"><a href="peak2genelinkage-with-archr.html#cb308-3"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb308-4"><a href="peak2genelinkage-with-archr.html#cb308-4"></a>    <span class="dt">addDOC =</span> <span class="ot">FALSE</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<blockquote>
<p>## NULL<br />
## NULL<br />
## NULL<br />
## NULL<br />
## NULL<br />
## NULL<br />
## NULL<br />
## NULL<br />
## NULL<br />
## [1] 0</p>
</blockquote>
<p><a href="images/HemeWalkthrough/PDF/Plot-Tracks-Marker-Genes-with-Peak2GeneLinks.pdf">Download PDF : Plot-Tracks-Marker-Genes.pdf</a></p>
</div>
<div id="plotting-a-heatmap-of-peak-to-gene-links" class="section level3" number="15.3.2">
<h3><span class="header-section-number">15.3.2</span> Plotting a heatmap of peak-to-gene links</h3>
<p>To visualize the correspondence of all of our peak-to-gene links, we can plot a peak-to-gene heatmap which contains two side-by-side heatmaps, one for our scATAC-seq data and one for our scRNA-seq data. To do this, we use the <code>plotPeak2GeneHeatmap()</code></p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="peak2genelinkage-with-archr.html#cb309-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotPeak2GeneHeatmap</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotPeak2GeneHeatmap-11a707e40d64f-Date-2020-04-15_Time-11-59-30.log<br />
## If there is an issue, please report to github with logFile!<br />
## 2020-04-15 11:59:34 : Determining KNN Groups!, 0.071 mins elapsed.<br />
## 2020-04-15 11:59:39 : Ordering Peak2Gene Links!, 0.15 mins elapsed.<br />
## 2020-04-15 11:59:48 : Constructing ATAC Heatmap!, 0.309 mins elapsed.<br />
## Adding Annotations..<br />
## Preparing Main Heatmap..<br />
## 2020-04-15 11:59:49 : Constructing RNA Heatmap!, 0.32 mins elapsed.<br />
## Adding Annotations..<br />
## Preparing Main Heatmap..<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotPeak2GeneHeatmap-11a707e40d64f-Date-2020-04-15_Time-11-59-30.log</p>
</blockquote>
<p>The heatmap rows are clustered using k-means clustering based on the value passed to the parameter <code>k</code>, which defaults to 25 as shown below.</p>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="peak2genelinkage-with-archr.html#cb310-1"></a>p</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Peak2GeneHeatmap_1.png" width="600" height="800" /></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="co-accessibility-with-archr.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="identification-of-positive-tf-regulators.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/15_IntegrativeAnalysis.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/15_IntegrativeAnalysis.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
},
"search": true,
"info": false
});
});
</script>

</body>

</html>
